From 9dc2c365375386f06a919456dac9ffbe6849c86e Mon Sep 17 00:00:00 2001
From: Richard Lowe <richlowe@richlowe.net>
Date: Tue, 4 Mar 2014 22:11:03 +0000
Subject: [PATCH 08/12] sol2: enable full __cxa_atexit support

diff -wpruN '--exclude=*.orig' gcc-8.1.0~/gcc/config/sol2.h gcc-8.1.0/gcc/config/sol2.h
--- gcc-8.1.0~/gcc/config/sol2.h	2018-01-30 21:18:40.000000000 +0000
+++ gcc-8.1.0/gcc/config/sol2.h	2018-05-24 11:10:28.740109324 +0000
@@ -206,7 +206,7 @@ along with GCC; see the file COPYING3.
 				   shared|" PIE_SPEC ":crtbeginS.o%s; \
 				   :crtbegin.o%s}"
 #else
-#define STARTFILE_CRTBEGIN_SPEC	"crtbegin.o%s"
+#define STARTFILE_CRTBEGIN_SPEC	"%{shared:crtbeginS.o%s;:crtbegin.o%s}"
 #endif
 
 #if ENABLE_VTABLE_VERIFY
@@ -256,7 +256,7 @@ along with GCC; see the file COPYING3.
 			       shared|" PIE_SPEC ":crtendS.o%s; \
 			       :crtend.o%s}"
 #else
-#define ENDFILE_CRTEND_SPEC "crtend.o%s"
+#define ENDFILE_CRTEND_SPEC "%{shared:crtendS.o%s;:crtend.o%s}"
 #endif
 
 #undef  ENDFILE_SPEC
diff -wpruN '--exclude=*.orig' gcc-8.1.0~/libgcc/config.host gcc-8.1.0/libgcc/config.host
--- gcc-8.1.0~/libgcc/config.host	2018-04-06 20:04:17.000000000 +0000
+++ gcc-8.1.0/libgcc/config.host	2018-05-24 11:10:28.740693416 +0000
@@ -267,7 +267,7 @@ case ${host} in
 *-*-solaris2*)
   # Unless linker support and dl_iterate_phdr are present,
   # unwind-dw2-fde-dip.c automatically falls back to unwind-dw2-fde.c.
-  tmake_file="$tmake_file sol2/t-sol2 t-eh-dw2-dip t-crtstuff-pic t-libgcc-pic t-slibgcc t-slibgcc-elf-ver"
+  tmake_file="$tmake_file sol2/t-sol2 t-eh-dw2-dip t-crtstuff-pic t-libgcc-pic t-slibgcc t-slibgcc-elf-ver t-crtstuff-pic"
   if test $with_gnu_ld = yes; then
     tmake_file="$tmake_file t-slibgcc-gld"
   else
@@ -287,6 +287,7 @@ case ${host} in
       i?86-*-solaris2.1[0-9]* | x86_64-*-solaris2.1[0-9]*)
         # Solaris 10+/x86 provides crt1.o, crti.o, crtn.o, and gcrt1.o as
         # part of the base system.
+        extra_parts="$extra_parts crtbeginS.o crtendS.o"
         ;;
       sparc*-*-solaris2.1[0-9]*)
         # Solaris 10+/SPARC lacks crt1.o and gcrt1.o.
