From da0a2bd04983b4442b739612dd65722f2f7147c1 Mon Sep 17 00:00:00 2001
From: Richard Lowe <richlowe@richlowe.net>
Date: Sun, 24 Jan 2016 18:30:00 +0000
Subject: [PATCH 09/12] symtab: disable non-interposable alias generation if
 function cloning is disabled

diff -wpruN '--exclude=*.orig' gcc-8.1.0~/gcc/symtab.c gcc-8.1.0/gcc/symtab.c
--- gcc-8.1.0~/gcc/symtab.c	2018-04-26 20:05:09.000000000 +0000
+++ gcc-8.1.0/gcc/symtab.c	2018-05-24 11:10:37.764222822 +0000
@@ -1764,6 +1764,10 @@ symtab_node::noninterposable_alias (void
   tree new_decl;
   symtab_node *new_node = NULL;
 
+  /* Do not allow a clone to be created if function-cloning is disabled */
+  if (!flag_clone_functions)
+	  return NULL;
+
   /* First try to look up existing alias or base object
      (if that is already non-overwritable).  */
   symtab_node *node = ultimate_alias_target ();
