From 7e51bee83d5a041b732fa59dae0694047144ebc3 Mon Sep 17 00:00:00 2001
From: Richard Lowe <richlowe@richlowe.net>
Date: Mon, 29 Oct 2018 18:23:00 +0000
Subject: [PATCH 16/17] libgo: libelf on illumos doesn't support largefile

Originally from pkgsrc
diff -wpruN '--exclude=*.orig' a~/libgo/runtime/proc.c a/libgo/runtime/proc.c
--- a~/libgo/runtime/proc.c	1970-01-01 00:00:00
+++ a/libgo/runtime/proc.c	1970-01-01 00:00:00
@@ -12,6 +12,10 @@
 #include "config.h"
 
 #ifdef HAVE_DL_ITERATE_PHDR
+#ifdef __sun
+#undef _FILE_OFFSET_BITS
+#define _FILE_OFFSET_BITS 32
+#endif
 #include <link.h>
 #endif
 
