diff -wpruN '--exclude=*.orig' gcc-8.1.0~/libstdc++-v3/acinclude.m4 gcc-8.1.0/libstdc++-v3/acinclude.m4
--- gcc-8.1.0~/libstdc++-v3/acinclude.m4	2017-11-21 06:22:13.000000000 +0000
+++ gcc-8.1.0/libstdc++-v3/acinclude.m4	2018-05-24 11:11:23.704129435 +0000
@@ -1434,7 +1434,7 @@ AC_DEFUN([GLIBCXX_ENABLE_LIBSTDCXX_TIME]
         ac_has_nanosleep=yes
         ;;
       solaris*)
-        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
+#        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
         ac_has_clock_monotonic=yes
         ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
diff -wpruN '--exclude=*.orig' gcc-8.1.0~/libstdc++-v3/configure gcc-8.1.0/libstdc++-v3/configure
--- gcc-8.1.0~/libstdc++-v3/configure	2018-04-24 16:45:26.000000000 +0000
+++ gcc-8.1.0/libstdc++-v3/configure	2018-05-24 11:11:23.716474110 +0000
@@ -20581,7 +20581,7 @@ $as_echo "$glibcxx_glibc217" >&6; }
         ac_has_nanosleep=yes
         ;;
       solaris*)
-        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
+#        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
         ac_has_clock_monotonic=yes
         ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
