diff -wpruN '--exclude=*.orig' a~/lib/freebl/blinit.c a/lib/freebl/blinit.c
--- a~/lib/freebl/blinit.c	1970-01-01 00:00:00
+++ a/lib/freebl/blinit.c	1970-01-01 00:00:00
@@ -131,7 +131,7 @@ CheckX86CPUSupport()
 #endif /* NSS_X86_OR_X64 */
 
 /* clang-format off */
-#if (defined(__aarch64__) || defined(__arm__)) && !defined(TARGET_OS_IPHONE)
+#if (defined(__aarch64__) || defined(__arm__)) && (!defined(TARGET_OS_IPHONE) && !defined(__sun))
 #ifndef __has_include
 #define __has_include(x) 0
 #endif
@@ -169,7 +169,7 @@ static unsigned long getauxval(unsigned
 #define AT_HWCAP 16
 #endif
 
-#endif /* defined(__aarch64__) || defined(__arm__) */
+#endif /* (defined(__aarch64__) || defined(__arm__)) && !defined(__sun) */
 /* clang-format on */
 
 #if defined(__aarch64__)
@@ -544,7 +544,7 @@ FreeblInit(void)
 {
 #ifdef NSS_X86_OR_X64
     CheckX86CPUSupport();
-#elif (defined(__aarch64__) || defined(__arm__))
+#elif (defined(__aarch64__) || defined(__arm__)) && !defined(__sun)
     CheckARMSupport();
 #elif (defined(__powerpc__))
     CheckPPCSupport();
diff -wpruN '--exclude=*.orig' a~/lib/freebl/gcm.c a/lib/freebl/gcm.c
--- a~/lib/freebl/gcm.c	1970-01-01 00:00:00
+++ a/lib/freebl/gcm.c	1970-01-01 00:00:00
@@ -18,6 +18,8 @@
 
 #include <limits.h>
 
+/* XXXARM: This ICE's the compiler, sigh */
+#ifndef __aarch64__
 /* old gcc doesn't support some poly64x2_t intrinsic */
 #if defined(__aarch64__) && defined(IS_LITTLE_ENDIAN) && \
     (defined(__clang__) || defined(__GNUC__) && __GNUC__ > 6)
@@ -27,6 +29,7 @@
 /* We don't test on big endian platform, so disable this on big endian. */
 #define USE_ARM_GCM
 #endif
+#endif
 
 /* Forward declarations */
 SECStatus gcm_HashInit_hw(gcmHashContext *ghash);
diff -wpruN '--exclude=*.orig' a~/lib/libpkix/pkix/util/pkix_tools.c a/lib/libpkix/pkix/util/pkix_tools.c
--- a~/lib/libpkix/pkix/util/pkix_tools.c	1970-01-01 00:00:00
+++ a/lib/libpkix/pkix/util/pkix_tools.c	1970-01-01 00:00:00
@@ -479,7 +479,7 @@ pkix_hex2i(char c)
 char
 pkix_i2hex(char digit)
 {
-        if ((digit >= 0)&&(digit <= 9))
+        if (((signed char)digit >= 0)&&(digit <= 9))
                 return (digit+'0');
         else if ((digit >= 0xa)&&(digit <= 0xf))
                 return (digit - 10 + 'a');
diff -wpruN '--exclude=*.orig' a~/lib/pk11wrap/pk11hpke.c a/lib/pk11wrap/pk11hpke.c
--- a~/lib/pk11wrap/pk11hpke.c	1970-01-01 00:00:00
+++ a/lib/pk11wrap/pk11hpke.c	1970-01-01 00:00:00
@@ -16,7 +16,6 @@
 #include "secmod.h"
 #include "secmodi.h"
 #include "secmodti.h"
-#include "secutil.h"
 
 #define SERIALIZATION_VERSION 2
 
diff -wpruN '--exclude=*.orig' a~/lib/pk11wrap/pk11pub.h a/lib/pk11wrap/pk11pub.h
--- a~/lib/pk11wrap/pk11pub.h	1970-01-01 00:00:00
+++ a/lib/pk11wrap/pk11pub.h	1970-01-01 00:00:00
@@ -847,12 +847,14 @@ SECStatus PK11_AEADRawOp(PK11Context *co
                          unsigned char *out, int *outlen,
                          int maxout, const unsigned char *in, int inlen);
 /* NSS builds the mechanism specific params */
+#if CRYPTOKI_VERSION_MAJOR >= 3
 SECStatus PK11_AEADOp(PK11Context *context, CK_ULONG ivGen,
                       int fixedbits, unsigned char *iv, int ivlen,
                       const unsigned char *aad, int aadlen,
                       unsigned char *out, int *outlen,
                       int maxout, unsigned char *tag, int taglen,
                       const unsigned char *in, int inlen);
+#endif
 
 SECStatus PK11_Finalize(PK11Context *context);
 SECStatus PK11_DigestFinal(PK11Context *context, unsigned char *data,
