https://bugzilla.mozilla.org/show_bug.cgi?id=1819022

diff -wpruN '--exclude=*.orig' a~/coreconf/config.gypi a/coreconf/config.gypi
--- a~/coreconf/config.gypi	1970-01-01 00:00:00
+++ a/coreconf/config.gypi	1970-01-01 00:00:00
@@ -170,7 +170,7 @@
           'NSS_NO_INIT_SUPPORT',
         ],
       }],
-      [ 'OS!="android" and OS!="mac" and OS!="ios" and OS!="win"', {
+      [ 'OS!="android" and OS!="mac" and OS!="ios" and OS!="win" and OS!="solaris"', {
         'libraries': [
           '-lpthread',
         ],
@@ -278,6 +278,9 @@
             'sources': [
               '<(INTERMEDIATE_DIR)/out.>(mapfile)',
             ],
+            'ldflags': [
+              '-Wl,-M,<(INTERMEDIATE_DIR)/out.>(mapfile)',
+            ],
             'actions': [{
               'action_name': 'generate_mapfile',
               'inputs': [
@@ -412,6 +415,11 @@
               '-z', 'noexecstack',
             ],
           }],
+          ['OS=="solaris"', {
+            'ldflags': [
+              '-z', 'combreloc', '-z', 'defs', '-z', 'ignore',
+            ],
+          }],
           [ 'OS!="mac" and OS!="ios" and OS!="win"', {
             'cflags': [
               '-fPIC',
diff -wpruN '--exclude=*.orig' a~/lib/freebl/freebl.gyp a/lib/freebl/freebl.gyp
--- a~/lib/freebl/freebl.gyp	1970-01-01 00:00:00
+++ a/lib/freebl/freebl.gyp	1970-01-01 00:00:00
@@ -83,7 +83,7 @@
           ],
         }],
         [ 'OS=="linux" or OS=="android" or OS=="dragonfly" or OS=="freebsd" or \
-           OS=="netbsd" or OS=="openbsd"', {
+           OS=="netbsd" or OS=="openbsd" or OS=="solaris"', {
           'cflags': [
             '-mpclmul',
             '-maes',
@@ -150,7 +150,7 @@
           ],
         }],
         [ 'OS=="linux" or OS=="android" or OS=="dragonfly" or OS=="freebsd" or \
-           OS=="netbsd" or OS=="openbsd"', {
+           OS=="netbsd" or OS=="openbsd" or OS=="solaris"', {
           'cflags': [
             '-mpclmul',
             '-maes',
@@ -204,7 +204,7 @@
         '-mpclmul', '-maes'
       ],
       'conditions': [
-        [ 'OS=="linux" or OS=="android" or OS=="dragonfly" or OS=="freebsd" or OS=="netbsd" or OS=="openbsd"', {
+        [ 'OS=="linux" or OS=="android" or OS=="dragonfly" or OS=="freebsd" or OS=="netbsd" or OS=="openbsd" or OS=="solaris"', {
           'cflags': [
             '-mpclmul', '-maes'
           ],
