diff --git a/lib/freebl/freebl.gyp b/lib/freebl/freebl.gyp
index 72e1b9b..1d1ba74 100644
--- a/lib/freebl/freebl.gyp
+++ b/lib/freebl/freebl.gyp
@@ -83,7 +83,7 @@
           ],
         }],
         [ 'OS=="linux" or OS=="android" or OS=="dragonfly" or OS=="freebsd" or \
-           OS=="netbsd" or OS=="openbsd"', {
+           OS=="netbsd" or OS=="openbsd" or OS=="solaris"', {
           'cflags': [
             '-mpclmul',
             '-maes',
@@ -150,7 +150,7 @@
           ],
         }],
         [ 'OS=="linux" or OS=="android" or OS=="dragonfly" or OS=="freebsd" or \
-           OS=="netbsd" or OS=="openbsd"', {
+           OS=="netbsd" or OS=="openbsd" or OS=="solaris"', {
           'cflags': [
             '-mpclmul',
             '-maes',
@@ -204,7 +204,7 @@
         '-mpclmul', '-maes'
       ],
       'conditions': [
-        [ 'OS=="linux" or OS=="android" or OS=="dragonfly" or OS=="freebsd" or OS=="netbsd" or OS=="openbsd"', {
+        [ 'OS=="linux" or OS=="android" or OS=="dragonfly" or OS=="freebsd" or OS=="netbsd" or OS=="openbsd" or OS=="solaris"', {
           'cflags': [
             '-mpclmul', '-maes'
           ],
