--- a/Makefile.shared~	2023-03-22 22:12:09.028331656 +0000
+++ a/Makefile.shared	2023-03-22 22:12:47.760970959 +0000
@@ -362,7 +362,7 @@
 	$(LINK_APP)
 
 link_o.solaris:
-	@ if $(DETECT_GNU_LD); then \
+	if $(DETECT_GNU_LD); then \
 		$(DO_GNU_SO); \
 	else \
 		$(CALC_VERSIONS); \
@@ -376,7 +376,7 @@
 	fi; \
 	$(LINK_SO_O)
 link_a.solaris:
-	@ if $(DETECT_GNU_LD); then \
+	if $(DETECT_GNU_LD); then \
 		$(DO_GNU_SO); \
 	else \
 		$(CALC_VERSIONS); \
@@ -390,7 +390,7 @@
 	fi; \
 	$(LINK_SO_A)
 link_app.solaris:
-	@ if $(DETECT_GNU_LD); then \
+	if $(DETECT_GNU_LD); then \
 		$(DO_GNU_APP); \
 	else \
 		LDFLAGS="$(CFLAGS) -R $(LIBRPATH)"; \
@@ -577,7 +577,7 @@
 # Targets to build symbolic links when needed
 symlink.gnu symlink.solaris symlink.svr3 symlink.svr5 symlink.irix \
 symlink.aix symlink.reliantunix:
-	@ $(CALC_VERSIONS); \
+	$(CALC_VERSIONS); \
 	SHLIB=lib$(LIBNAME).so; \
 	$(SYMLINK_SO)
 symlink.darwin:
