From f8b398433b252bde0702c85106f3ea6000cfff3d Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <omnios@citrus-it.co.uk>
Date: Thu, 9 May 2019 13:43:30 +0000
Subject: OOCE: Adjust default library paths for OmniOS

---
 gcc/config/sol2.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/gcc/config/sol2.h b/gcc/config/sol2.h
index 9b4ae3ae3f9..a4a72718b75 100644
--- a/gcc/config/sol2.h
+++ b/gcc/config/sol2.h
@@ -306,8 +306,8 @@ along with GCC; see the file COPYING3.  If not see
   "%{G:-G} \
    %{YP,*} \
    %{R*} \
-   %{!YP,*:%{p|pg:-Y P,%R/usr/lib/libp%R/lib:%R/usr/lib} \
-	   %{!p:%{!pg:-Y P,%R/lib:%R/usr/lib}}}"
+   %{!YP,*:%{p|pg:-Y P,%R/usr/gcc/9/lib:%R/lib:%R/usr/lib -R %R/usr/gcc/9/lib -L %R/usr/gcc/9/lib} \
+	   %{!p:%{!pg:-Y P,%R/usr/gcc/9/lib:%R/lib:%R/usr/lib -R %R/usr/gcc/9/lib -L %R/usr/gcc/9/lib}}}"
 
 #undef LINK_ARCH32_SPEC
 #define LINK_ARCH32_SPEC LINK_ARCH32_SPEC_BASE
@@ -319,8 +319,8 @@ along with GCC; see the file COPYING3.  If not see
   "%{G:-G} \
    %{YP,*} \
    %{R*} \
-   %{!YP,*:%{p|pg:-Y P,%R/usr/lib/libp/" ARCH64_SUBDIR ":%R/lib/" ARCH64_SUBDIR ":%R/usr/lib/" ARCH64_SUBDIR "}	\
-	   %{!p:%{!pg:-Y P,%R/lib/" ARCH64_SUBDIR ":%R/usr/lib/" ARCH64_SUBDIR "}}}"
+   %{!YP,*:%{p|pg:-Y P,%R/usr/gcc/9/lib/" ARCH64_SUBDIR ":%R/lib/" ARCH64_SUBDIR ":%R/usr/lib/" ARCH64_SUBDIR " -R %R/usr/gcc/9/lib/" ARCH64_SUBDIR " -L %R/usr/gcc/9/lib/" ARCH64_SUBDIR "}	\
+	   %{!p:%{!pg:-Y P,%R/usr/gcc/9/lib/" ARCH64_SUBDIR ":%R/lib/" ARCH64_SUBDIR ":%R/usr/lib/" ARCH64_SUBDIR " -R %R/usr/gcc/9/lib/" ARCH64_SUBDIR " -L %R/usr/gcc/9/lib/" ARCH64_SUBDIR "}}}"
 
 #undef LINK_ARCH64_SPEC
 #ifndef USE_GLD
-- 
2.23.0

