diff -wpruN '--exclude=*.orig' a~/filan.c a/filan.c
--- a~/filan.c	1970-01-01 00:00:00
+++ a/filan.c	1970-01-01 00:00:00
@@ -30,7 +30,7 @@ static int filan_streams_analyze(int fd,
 
 /* dirty workaround so we dont get an error on AIX when being linked with
    libwrap */
-int allow_severity, deny_severity;
+extern int allow_severity, deny_severity;
 
 /* global variables for configuring filan */
 bool filan_followsymlinks;
