diff -wpruN '--exclude=*.orig' a~/include/ntp_config.h a/include/ntp_config.h
--- a~/include/ntp_config.h	1970-01-01 00:00:00
+++ a/include/ntp_config.h	1970-01-01 00:00:00
@@ -14,7 +14,7 @@
  */
 #ifndef CONFIG_FILE
 # ifndef SYS_WINNT
-#  define	CONFIG_FILE "/etc/ntp.conf"
+#  define	CONFIG_FILE "/etc/inet/ntp.conf"
 # else /* SYS_WINNT */
 #  define	CONFIG_FILE	"%windir%\\system32\\drivers\\etc\\ntp.conf"
 #  define	ALT_CONFIG_FILE "%windir%\\ntp.conf"
diff -wpruN '--exclude=*.orig' a~/ntpd/refclock_oncore.c a/ntpd/refclock_oncore.c
--- a~/ntpd/refclock_oncore.c	1970-01-01 00:00:00
+++ a/ntpd/refclock_oncore.c	1970-01-01 00:00:00
@@ -1211,7 +1211,7 @@ oncore_read_config(
 
 	FILE	*fd;
 	char	*cc, *ca, line[100], units[2], device[64];
-	const char	*dirs[] = { "/etc/ntp", "/etc", 0 };
+	const char	*dirs[] = { "/etc/inet", "/etc", 0 };
 	const char *cp, **cpp;
 	int	i, sign, lat_flg, long_flg, ht_flg, mode, mask;
 	double	f1, f2, f3;
