diff -wpruN '--exclude=*.orig' a~/sntp/libevent/evutil_rand.c a/sntp/libevent/evutil_rand.c
--- a~/sntp/libevent/evutil_rand.c	1970-01-01 00:00:00
+++ a/sntp/libevent/evutil_rand.c	1970-01-01 00:00:00
@@ -195,8 +195,15 @@ evutil_secure_rng_get_bytes(void *buf, s
 void
 evutil_secure_rng_add_bytes(const char *buf, size_t n)
 {
+	/*
+	 * Apparently OpenBSD has an arc4random_addrandom(), which illumos
+	 * does not.  Since this patch is for a hacky static-only instantiation
+	 * of libevent, we just if 0 this out.  :(
+	 */
+#if 0
 	arc4random_addrandom((unsigned char*)buf,
 	    n>(size_t)INT_MAX ? INT_MAX : (int)n);
+#endif
 }
 
 void
