diff -wpruN '--exclude=*.orig' a~/ntpd/ntp_proto.c a/ntpd/ntp_proto.c
--- a~/ntpd/ntp_proto.c	1970-01-01 00:00:00
+++ a/ntpd/ntp_proto.c	1970-01-01 00:00:00
@@ -2766,6 +2766,16 @@ clock_update(
 	char	*fmri;
 #endif /* HAVE_LIBSCF_H */
 
+#ifdef DEBUG_TIMING
+{
+        char buf[2048];
+
+        snprintf(buf, sizeof(buf), "%lu %lu %d %d epoch",
+                 sys_epoch,peer->epoch,peer->epoch-sys_epoch,current_time - peer->epoch);
+        record_timing_stats(buf);
+}
+#endif
+
 	/*
 	 * Update the system state variables. We do this very carefully,
 	 * as the poll interval might need to be clamped differently.
