diff -wpruN '--exclude=*.orig' a~/aclocal.m4 a/aclocal.m4
--- a~/aclocal.m4	1970-01-01 00:00:00
+++ a/aclocal.m4	1970-01-01 00:00:00
@@ -534,8 +534,10 @@ AC_DEFUN(AC_LBL_SHLIBS_INIT,
 		    # XXX - this assumes GCC is using the Sun linker,
 		    # rather than the GNU linker.
 		    #
-		    V_SONAME_OPT="-Wl,-h,"
-		    V_RPATH_OPT="-Wl,-R,"
+                    V_SHLIB_CMD="\$(LD)"
+                    V_SHLIB_OPT="-G -zignore"
+                    V_SONAME_OPT="-h "
+                    V_RPATH_OPT="-R"
 		    ;;
 	    esac
     else
