diff -wpruN '--exclude=*.orig' a~/tools/plot/fio2gnuplot a/tools/plot/fio2gnuplot
--- a~/tools/plot/fio2gnuplot	1970-01-01 00:00:00
+++ a/tools/plot/fio2gnuplot	1970-01-01 00:00:00
@@ -406,9 +406,9 @@ def main(argv):
     force_keep_temp_files=False
 
     if not os.path.isfile(gpm_dir+'math.gpm'):
-        gpm_dir="/usr/local/share/fio/"
+        gpm_dir="/usr/share/fio/"
         if not os.path.isfile(gpm_dir+'math.gpm'):
-            print("Looks like fio didn't get installed properly as no gpm files found in '/usr/share/fio' or '/usr/local/share/fio'\n")
+            print("Looks like fio didn't get installed properly as no gpm files found in '/usr/share/fio'\n")
             sys.exit(3)
 
     try:
