
Prevent unecessary linking with old filter libraries.

diff -wpruN '--exclude=*.orig' a~/Makefile a/Makefile
--- a~/Makefile	1970-01-01 00:00:00
+++ a/Makefile	1970-01-01 00:00:00
@@ -210,7 +210,7 @@ endif
   LDFLAGS += -rdynamic
 endif
 ifeq ($(CONFIG_TARGET_OS), SunOS)
-  LIBS	 += -lpthread -ldl
+  #LIBS	 += -lpthread -ldl
   CPPFLAGS += -D__EXTENSIONS__
 endif
 ifeq ($(CONFIG_TARGET_OS), FreeBSD)
diff -wpruN '--exclude=*.orig' a~/configure a/configure
--- a~/configure	1970-01-01 00:00:00
+++ a/configure	1970-01-01 00:00:00
@@ -780,7 +780,7 @@ int main(void)
 EOF
 if compile_prog "" "-laio" "solarisaio" ; then
   solaris_aio=yes
-  LIBS="-laio $LIBS"
+  #LIBS="-laio $LIBS"
 fi
 print_config "Solaris AIO support" "$solaris_aio"
 
