
Prevent unecessary linking with old filter libraries.

diff -wpruN '--exclude=*.orig' a~/Makefile a/Makefile
--- a~/Makefile	1970-01-01 00:00:00
+++ a/Makefile	1970-01-01 00:00:00
@@ -177,7 +177,7 @@ ifeq ($(CONFIG_TARGET_OS), Android)
   LDFLAGS += -rdynamic
 endif
 ifeq ($(CONFIG_TARGET_OS), SunOS)
-  LIBS	 += -lpthread -ldl
+  #LIBS	 += -lpthread -ldl
   CPPFLAGS += -D__EXTENSIONS__
 endif
 ifeq ($(CONFIG_TARGET_OS), FreeBSD)
diff -wpruN '--exclude=*.orig' a~/configure a/configure
--- a~/configure	1970-01-01 00:00:00
+++ a/configure	1970-01-01 00:00:00
@@ -609,7 +609,7 @@ EOF
   elif compile_prog "" "-laio" "libaio" ; then
     libaio=yes
     libaio_uring=no
-    LIBS="-laio $LIBS"
+    #LIBS="-laio $LIBS"
   else
     if test "$libaio" = "yes" ; then
       feature_not_found "linux AIO" "libaio-dev or libaio-devel"
@@ -761,7 +761,7 @@ int main(void)
 EOF
 if compile_prog "" "-laio" "solarisaio" ; then
   solaris_aio=yes
-  LIBS="-laio $LIBS"
+  #LIBS="-laio $LIBS"
 fi
 print_config "Solaris AIO support" "$solaris_aio"
 
