diff -wpruN '--exclude=*.orig' fio-fio-3.7~/Makefile fio-fio-3.7/Makefile
--- fio-fio-3.7~/Makefile	2018-06-01 21:21:56.000000000 +0000
+++ fio-fio-3.7/Makefile	2018-06-05 14:19:06.211331194 +0000
@@ -305,16 +305,19 @@ endif
 endif
 
 ifeq ($(CONFIG_TARGET_OS), SunOS)
-	INSTALL = ginstall
+	INSTALL = /usr/gnu/bin/install
 else
 	INSTALL = install
 endif
-prefix = $(INSTALL_PREFIX)
+prefix = /usr
 bindir = $(prefix)/bin
 
 ifeq ($(CONFIG_TARGET_OS), Darwin)
 mandir = /usr/share/man
 sharedir = /usr/share/fio
+else ifeq ($(CONFIG_TARGET_OS), SunOS)
+mandir = $(prefix)/share/man
+sharedir = $(prefix)/share/fio
 else
 mandir = $(prefix)/man
 sharedir = $(prefix)/share/fio
