diff -wpruN '--exclude=*.orig' a~/Makefile a/Makefile
--- a~/Makefile	1970-01-01 00:00:00
+++ a/Makefile	1970-01-01 00:00:00
@@ -415,11 +415,11 @@ endif
 endif
 
 ifeq ($(CONFIG_TARGET_OS), SunOS)
-	INSTALL = ginstall
+	INSTALL = /usr/gnu/bin/install
 else
 	INSTALL = install
 endif
-prefix = $(INSTALL_PREFIX)
+prefix = /usr
 bindir = $(prefix)/bin
 libdir = $(prefix)/lib/fio
 
@@ -427,7 +427,7 @@ ifeq ($(CONFIG_TARGET_OS), Darwin)
 mandir = /usr/share/man
 sharedir = /usr/share/fio
 else
-mandir = $(prefix)/man
+mandir = $(prefix)/share/man
 sharedir = $(prefix)/share/fio
 endif
 
