diff -wpruN '--exclude=*.orig' a~/tmux.h a/tmux.h
--- a~/tmux.h	1970-01-01 00:00:00
+++ a/tmux.h	1970-01-01 00:00:00
@@ -23,6 +23,12 @@
 #include <sys/uio.h>
 
 #include <event.h>
+#include <event_compat.h>
+#include <event_struct.h>
+#include <buffer.h>
+#include <bufferevent.h>
+#include <bufferevent_struct.h>
+#include <buffer_compat.h>
 #include <limits.h>
 #include <stdarg.h>
 #include <stdio.h>
