diff -wpruN '--exclude=*.orig' a~/lspci.man a/lspci.man
--- a~/lspci.man	1970-01-01 00:00:00
+++ a/lspci.man	1970-01-01 00:00:00
@@ -335,10 +335,7 @@ in a single record. Please avoid using t
 .SH FILES
 .TP
 .B @IDSDIR@/pci.ids
-A list of all known PCI ID's (vendors, devices, classes and subclasses). Maintained
-at https://pci-ids.ucw.cz/, use the
-.B update-pciids
-utility to download the most recent version.
+A list of all known PCI ID's (vendors, devices, classes and subclasses).
 .TP
 .B @IDSDIR@/pci.ids.gz
 If lspci is compiled with support for compression, this file is tried before pci.ids.
@@ -360,10 +357,7 @@ Access to the extended configuration spa
 back-end.
 
 .SH SEE ALSO
-.BR setpci (8),
-.BR pci.ids (5),
-.BR update-pciids (8),
-.BR pcilib (7)
+.BR setpci (8)
 
 .SH AUTHOR
 The PCI Utilities are maintained by Martin Mares <mj@ucw.cz>.
diff -wpruN '--exclude=*.orig' a~/setpci.man a/setpci.man
--- a~/setpci.man	1970-01-01 00:00:00
+++ a/setpci.man	1970-01-01 00:00:00
@@ -183,8 +183,7 @@ corresponds to the second word of the po
 asks for the first 32-bit word of the extended capability with ID 0x108.
 
 .SH SEE ALSO
-.BR lspci (8),
-.BR pcilib (7)
+.BR lspci (8)
 
 .SH AUTHOR
 The PCI Utilities are maintained by Martin Mares <mj@ucw.cz>.
