diff -wpruN '--exclude=*.orig' zip30~/unix/Makefile zip30/unix/Makefile
--- zip30~/unix/Makefile	2008-05-07 06:33:56.000000000 +0000
+++ zip30/unix/Makefile	2018-06-14 21:23:59.084923759 +0000
@@ -38,10 +38,10 @@ BINFLAGS = 755
 MANFLAGS = 644
 
 # target directories - where to install executables and man pages to
-prefix = /usr/local
+prefix = $(DESTDIR)$(PREFIX)
 BINDIR = $(prefix)/bin
 MANEXT=1
-MANDIR = $(prefix)/man/man$(MANEXT)
+MANDIR = $(prefix)/share/man/man$(MANEXT)
 ZIPMANUAL = zip.txt
 ZIPMANUALcloak = zipcloak.txt
 ZIPMANUALnote = zipnote.txt
