diff -wpruN '--exclude=*.orig' a~/crossenv/__init__.py a/crossenv/__init__.py
--- a~/crossenv/__init__.py	1970-01-01 00:00:00
+++ a/crossenv/__init__.py	1970-01-01 00:00:00
@@ -509,6 +509,8 @@ class CrossEnvBuilder(venv.EnvBuilder):
                 self.host_machine = self.host_gnu_type.split('-')[0]
 
         self.host_release = ''
+        if self.host_sysname == "sunos5":
+            self.host_release = '5.11'
         if self.macosx_deployment_target:
             try:
                 major, minor = self.macosx_deployment_target.split(".")
