diff -wpruN '--exclude=*.orig' a~/wafhelpers/asciidoc.py a/wafhelpers/asciidoc.py
--- a~/wafhelpers/asciidoc.py	1970-01-01 00:00:00
+++ a/wafhelpers/asciidoc.py	1970-01-01 00:00:00
@@ -37,8 +37,7 @@ def configure(ctx):
     # asciidoc versions 8.6.5-8.6.7 throw warnings for warp.adoc and versions
     # < 8.6.4 have no HTML5 backend.
     # asciidoc3 versions < 3.0.2 throw errors.
-    adoc_list = [['asciidoctor', (1, 5, 8)],
-                 ['asciidoc', (8, 6, 8)],
+    adoc_list = [['asciidoc', (8, 6, 8)],
                  ['asciidoc3', (3, 0, 2)],
                  ]
     for progname, asciidocminver in adoc_list:
