diff -wpruN '--exclude=*.orig' a~/include/ntpd.h a/include/ntpd.h
--- a~/include/ntpd.h	1970-01-01 00:00:00
+++ a/include/ntpd.h	1970-01-01 00:00:00
@@ -364,7 +364,7 @@ extern int	peer_associations;	/* mobiliz
  * System variables are declared here.	See Section 3.2 of the
  * specification.
  */
-int	sys_maxclock;		/* maximum candidates */
+extern int	sys_maxclock;		/* maximum candidates */
 struct system_variables {
 	uint8_t	sys_leap;		/* system leap indicator */
 	uint8_t	sys_stratum;		/* system stratum */
