diff -wpruN '--exclude=*.orig' a~/wscript a/wscript
--- a~/wscript	1970-01-01 00:00:00
+++ a/wscript	1970-01-01 00:00:00
@@ -265,8 +265,8 @@ def configure(ctx):
     # check for some libs first.  some options, like stack protector,
     # may depend on some libs, like -lssp
     ctx.check_cc(lib="m", comment="Math library")
-    ctx.check_cc(lib="rt", mandatory=False, comment="realtime library")
-    ctx.check_cc(lib="pthread", mandatory=False, comment="threads library")
+    #ctx.check_cc(lib="rt", mandatory=False, comment="realtime library")
+    #ctx.check_cc(lib="pthread", mandatory=False, comment="threads library")
     ctx.check_cc(lib="execinfo", mandatory=False,
                  comment="BSD backtrace library")
     ret = ctx.check_cc(lib="bsd", mandatory=False,
