diff -wpruN '--exclude=*.orig' net-snmp-5.8~/perl/agent/Makefile.PL net-snmp-5.8/perl/agent/Makefile.PL
--- net-snmp-5.8~/perl/agent/Makefile.PL	2018-07-16 14:33:40.000000000 +0000
+++ net-snmp-5.8/perl/agent/Makefile.PL	2018-07-24 09:51:36.908087118 +0000
@@ -95,7 +95,7 @@ sub AgentInitMakeParams {
                 " " . $Params{'LIBS'};
 	    $Params{'CCFLAGS'} = "-I../../include " . $Params{'CCFLAGS'};
 #	} else {
-	    $Params{'LIBS'} = `$opts->{'nsconfig'} --libdir` . $Params{'LIBS'};
+	    $Params{'LIBS'} = $Params{'LIBS'} . `$opts->{'nsconfig'} --libdir`;
 #	    $Params{'PREREQ_PM'} = {'NetSNMP::OID' => '0.1'};
 	}
 	$Params{'CCFLAGS'} =~ s/ -W(all|inline|strict-prototypes|write-strings|cast-qual|no-char-subscripts)//g; # ignore developer warnings
