diff -wpruN '--exclude=*.orig' net-snmp-5.8~/agent/mibgroup/kernel_sunos5.c net-snmp-5.8/agent/mibgroup/kernel_sunos5.c
--- net-snmp-5.8~/agent/mibgroup/kernel_sunos5.c	2018-07-24 09:51:41.463341555 +0000
+++ net-snmp-5.8/agent/mibgroup/kernel_sunos5.c	2018-07-24 09:51:41.879550292 +0000
@@ -310,7 +310,7 @@ getKstatInt(const char *classname, const
 	goto Return;
     }
    
-    if ((ks = kstat_lookup(ksc, statname, 0, "link")) != NULL)
+    if ((ks = kstat_lookup(ksc, statname, -1, "link")) != NULL)
     {
         kid = kstat_read(ksc, ks, NULL);
         if (kid == -1) {
