diff -wpruN '--exclude=*.orig' a~/DEFAULTS/Defaults.sunos5 a/DEFAULTS/Defaults.sunos5
--- a~/DEFAULTS/Defaults.sunos5	1970-01-01 00:00:00
+++ a/DEFAULTS/Defaults.sunos5	1970-01-01 00:00:00
@@ -32,7 +32,8 @@ DEFINCDIRS=	$(SRCROOT)/include
 DEFOSINCDIRS=
 LDPATH=		-L/opt/schily/lib
 #RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib -R$(OLIBSDIR)
-RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+#RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+#RUNPATH=	-R$(INS_BASE)/lib
 
 ###########################################################################
 #
diff -wpruN '--exclude=*.orig' a~/RULES/i86pc-sunos5-gcc.rul a/RULES/i86pc-sunos5-gcc.rul
--- a~/RULES/i86pc-sunos5-gcc.rul	1970-01-01 00:00:00
+++ a/RULES/i86pc-sunos5-gcc.rul	1970-01-01 00:00:00
@@ -27,7 +27,7 @@ include $(SRCROOT)/$(RULESDIR)/cc-$(C_AR
 OSDEFS +=	-D__SVR4 -DSVR4
 KDEFINES=	-DKERNEL -D_KERNEL
 
-LIB_SOCKET=	-lsocket -lnsl -ldl
+LIB_SOCKET=	-lsocket -lnsl
 LIB_MATH=	-lm
 LIB_KVM=	-lkvm -lelf
 LIB_VOLMGT=	-lvolmgt
