diff -ru cdrtools-3.01~/DEFAULTS/Defaults.sunos5 cdrtools-3.01/DEFAULTS/Defaults.sunos5
--- cdrtools-3.01~/DEFAULTS/Defaults.sunos5	2013-11-04 20:58:21.000000000 +0000
+++ cdrtools-3.01/DEFAULTS/Defaults.sunos5	2020-03-20 00:22:50.629525017 +0000
@@ -32,7 +32,8 @@
 DEFOSINCDIRS=
 LDPATH=		-L/opt/schily/lib
 #RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib -R$(OLIBSDIR)
-RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+#RUNPATH=	-R$(INS_BASE)/lib -R/opt/schily/lib
+#RUNPATH=	-R$(INS_BASE)/lib
 
 ###########################################################################
 #
diff -ru cdrtools-3.01~/RULES/i86pc-sunos5-gcc.rul cdrtools-3.01/RULES/i86pc-sunos5-gcc.rul
--- cdrtools-3.01~/RULES/i86pc-sunos5-gcc.rul	2014-03-28 16:43:53.000000000 +0000
+++ cdrtools-3.01/RULES/i86pc-sunos5-gcc.rul	2020-03-20 00:25:25.059420019 +0000
@@ -27,7 +27,7 @@
 OSDEFS +=	-D__SVR4 -DSVR4
 KDEFINES=	-DKERNEL -D_KERNEL
 
-LIB_SOCKET=	-lsocket -lnsl -ldl
+LIB_SOCKET=	-lsocket -lnsl
 LIB_MATH=	-lm
 LIB_KVM=	-lkvm -lelf
 LIB_VOLMGT=	-lvolmgt
