diff -wpruN '--exclude=*.orig' a~/m4/configmake.m4 a/m4/configmake.m4
--- a~/m4/configmake.m4	1970-01-01 00:00:00
+++ a/m4/configmake.m4	1970-01-01 00:00:00
@@ -29,7 +29,7 @@ AC_DEFUN([gl_CONFIGMAKE_PREP],
   AC_SUBST([pkgdatadir], ['${datadir}/${PACKAGE}'])
   AC_SUBST([pkgincludedir], ['${includedir}/${PACKAGE}'])
   AC_SUBST([pkglibdir], ['${libdir}/${PACKAGE}'])
-  AC_SUBST([pkglibexecdir], ['${libexecdir}/${PACKAGE}'])
+  AC_SUBST([pkglibexecdir], ['${libexecdir}'])
 ])
 
 # gl_CONFIGMAKE
