diff -wpruN '--exclude=*.orig' a~/m4/threadlib.m4 a/m4/threadlib.m4
--- a~/m4/threadlib.m4	1970-01-01 00:00:00
+++ a/m4/threadlib.m4	1970-01-01 00:00:00
@@ -241,6 +241,7 @@ AC_DEFUN([gl_PTHREADLIB_BODY],
             [gl_pthread_in_glibc=yes],
             [])
           ;;
+        solaris*) gl_pthread_in_glibc=yes ;;
       esac
       echo "$as_me:__oline__: gl_pthread_in_glibc=$gl_pthread_in_glibc" >&AS_MESSAGE_LOG_FD
 
