diff -wpruN '--exclude=*.orig' a~/src/who.c a/src/who.c
--- a~/src/who.c	1970-01-01 00:00:00
+++ a/src/who.c	1970-01-01 00:00:00
@@ -518,8 +518,8 @@ print_runlevel (const STRUCT_UTMP *utmp_
   unsigned char curr = UT_PID (utmp_ent) % 256;
 
   if (!runlevline)
-    runlevline = xmalloc (strlen (_("run-level")) + 3);
-  sprintf (runlevline, "%s %c", _("run-level"), curr);
+    runlevline = xmalloc (sizeof (utmp_ent->ut_line));
+  sprintf (runlevline, "%s", utmp_ent->ut_line);
 
   if (!comment)
     comment = xmalloc (strlen (_("last=")) + 2);
