patch taken from https://github.com/OpenIndiana/oi-userland

diff -wpruN '--exclude=*.orig' a~/css.c a/css.c
--- a~/css.c	1970-01-01 00:00:00
+++ a/css.c	1970-01-01 00:00:00
@@ -73,10 +73,11 @@ static CssParserState parseCssDeclaratio
   {
     if( (int) *cp == '\0' )
     { 
+      /* assign position to the end of line. */
+      *position = cp;
       cp = fileReadLine ();
       if( cp == NULL ){
         makeCssSimpleTag(name, kind, TRUE);
-        *position = cp;
         return P_STATE_AT_END;
       }
     }
