diff -wpruN '--exclude=*.orig' net-snmp-5.8~/agent/mibgroup/mibII.h net-snmp-5.8/agent/mibgroup/mibII.h
--- net-snmp-5.8~/agent/mibgroup/mibII.h	2018-07-16 14:33:40.000000000 +0000
+++ net-snmp-5.8/agent/mibgroup/mibII.h	2018-07-24 09:51:35.283930004 +0000
@@ -55,3 +55,10 @@ config_require(ip-mib/ipSystemStatsTable
  * ip-mib/ipv6ScopeZoneIndexTable
  */
 #endif
+
+/*
+ * For Solaris, enable additional tables.
+ */
+#if defined( solaris2 )
+config_require(ip-mib/ipSystemStatsTable ip-mib/ipAddressTable)
+#endif
