diff -wpruN '--exclude=*.orig' net-snmp-5.8~/agent/mibgroup/ucd-snmp/vmstat_solaris2.c net-snmp-5.8/agent/mibgroup/ucd-snmp/vmstat_solaris2.c
--- net-snmp-5.8~/agent/mibgroup/ucd-snmp/vmstat_solaris2.c	2018-07-16 14:33:40.000000000 +0000
+++ net-snmp-5.8/agent/mibgroup/ucd-snmp/vmstat_solaris2.c	2018-07-24 09:51:37.306934603 +0000
@@ -534,6 +534,13 @@ update_stats(unsigned int registrationNu
                     "time_diff: %lld\n", time_diff));
 
         /*
+		 * Assign a minimum value to get around divide by zero case
+		 */
+		if (time_diff == 0) {
+			time_diff = 1;
+		}
+
+        /*
          * swapin and swapout are in pages, MIB wants kB/s,so we just need to get kB and seconds 
          * For the others we need to get value per second 
          * getpagesize() returns pagesize in bytes 
