diff -wpruN '--exclude=*.orig' net-snmp-5.8~/agent/mibgroup/ucd-snmp/vmstat_solaris2.c net-snmp-5.8/agent/mibgroup/ucd-snmp/vmstat_solaris2.c
--- net-snmp-5.8~/agent/mibgroup/ucd-snmp/vmstat_solaris2.c	2018-07-24 09:51:37.306934603 +0000
+++ net-snmp-5.8/agent/mibgroup/ucd-snmp/vmstat_solaris2.c	2018-07-24 09:51:37.720832829 +0000
@@ -598,6 +598,13 @@ update_stats(unsigned int registrationNu
         }
 
         /*
+	 * Assign a minimum value to get around divide by zero case
+	 */
+	if (cpu_sum == 0) {
+		cpu_sum = 1;
+	}
+
+        /*
          * Now calculate the absolute percentage values 
          * Looks somewhat complicated sometimes but tries to get around using floats to increase speed 
          */
