diff -wpruN '--exclude=*.orig' net-snmp-5.8~/agent/mibgroup/kernel_sunos5.c net-snmp-5.8/agent/mibgroup/kernel_sunos5.c
--- net-snmp-5.8~/agent/mibgroup/kernel_sunos5.c	2018-07-24 09:51:38.542255684 +0000
+++ net-snmp-5.8/agent/mibgroup/kernel_sunos5.c	2018-07-24 09:51:38.959015465 +0000
@@ -90,7 +90,7 @@ mibcache        Mibcache[MIBCACHE_SIZE+1
      0},
     {MIB_IP_ROUTE, 200 * sizeof(mib2_ipRouteEntry_t), (void *) -1, 0, 30,
      0, 0},
-    {MIB_IP_NET, 100 * sizeof(mib2_ipNetToMediaEntry_t), (void *) -1, 0,
+    {MIB_IP_NET, 400 * sizeof(mib2_ipNetToMediaEntry_t), (void *) -1, 0,
      300, 0, 0},
     {MIB_ICMP, sizeof(mib2_icmp_t), (void *) -1, 0, 60, 0, 0},
     {MIB_TCP, sizeof(mib2_tcp_t), (void *) -1, 0, 60, 0, 0},
