diff -wpruN '--exclude=*.orig' net-snmp-5.8~/agent/mibgroup/kernel_sunos5.c net-snmp-5.8/agent/mibgroup/kernel_sunos5.c
--- net-snmp-5.8~/agent/mibgroup/kernel_sunos5.c	2018-07-24 09:51:44.648186134 +0000
+++ net-snmp-5.8/agent/mibgroup/kernel_sunos5.c	2018-07-24 09:51:45.041785376 +0000
@@ -62,7 +62,6 @@
 
 #include <libdladm.h>
 #include <libdllink.h>
-#include <libnetcfg.h>
 /*-
  * Includes of local application header files 
  */
