diff -wpruN '--exclude=*.orig' net-snmp-5.8~/configure net-snmp-5.8/configure
--- net-snmp-5.8~/configure	2018-07-16 14:33:50.000000000 +0000
+++ net-snmp-5.8/configure	2018-07-24 09:51:45.439837943 +0000
@@ -12886,7 +12886,7 @@ lt_prog_compiler_static=
       ;;
 
     solaris*)
-      lt_prog_compiler_pic='-KPIC'
+      lt_prog_compiler_pic='-fPIC'
       lt_prog_compiler_static='-Bstatic'
       case $cc_basename in
       f77* | f90* | f95* | sunf77* | sunf90* | sunf95*)
