see: https://www.illumos.org/issues/14418

diff -wpruN '--exclude=*.orig' a~/src/hotspot/os/solaris/os_solaris.cpp a/src/hotspot/os/solaris/os_solaris.cpp
--- a~/src/hotspot/os/solaris/os_solaris.cpp	1970-01-01 00:00:00
+++ a/src/hotspot/os/solaris/os_solaris.cpp	1970-01-01 00:00:00
@@ -167,12 +167,6 @@ address os::Solaris::_main_stack_base =
 
 os::Solaris::pthread_setname_np_func_t os::Solaris::_pthread_setname_np = NULL;
 
-// "default" initializers for missing libc APIs
-extern "C" {
-  int memcntl(caddr_t, size_t, int, caddr_t, int, int);
-  int meminfo(const uint64_t *, int, const uint_t *, int, uint64_t *, uint_t *);
-}
-
 // "default" initializers for pthread-based synchronization
 extern "C" {
   static int pthread_mutex_default_init(mutex_t *mx, int scope, void *arg) { memset(mx, 0, sizeof(mutex_t)); return 0; }
