diff -wpruN '--exclude=*.orig' a~/src/include/tcsd.h a/src/include/tcsd.h
--- a~/src/include/tcsd.h	1970-01-01 00:00:00
+++ a/src/include/tcsd.h	1970-01-01 00:00:00
@@ -175,8 +175,8 @@ void	   thread_signal_init();
 
 /* signal handling */
 #ifndef __APPLE__
-struct sigaction tcsd_sa_int;
-struct sigaction tcsd_sa_chld;
+/*extern struct sigaction tcsd_sa_int;*/
+/*extern struct sigaction tcsd_sa_chld;*/
 #endif
 
 #endif
