From 56d22cc2f87c8d64b5bd4460374728e0c84c0f7c Mon Sep 17 00:00:00 2001
From: Richard Lowe <richlowe@richlowe.net>
Date: Tue, 4 Mar 2014 22:11:03 +0000
Subject: [PATCH 08/11] sol2: enable full __cxa_atexit support

diff -prwuN '--exclude=*.orig' gcc-7.3.0~/gcc/config/sol2.h gcc-7.3.0/gcc/config/sol2.h
--- gcc-7.3.0~/gcc/config/sol2.h	2017-11-21 09:31:12.135035000 +0000
+++ gcc-7.3.0/gcc/config/sol2.h	2018-05-08 14:44:26.269107565 +0000
@@ -178,7 +178,7 @@ along with GCC; see the file COPYING3.
 				   shared|" PIE_SPEC ":crtbeginS.o%s; \
 				   :crtbegin.o%s}"
 #else
-#define STARTFILE_CRTBEGIN_SPEC	"crtbegin.o%s"
+#define STARTFILE_CRTBEGIN_SPEC	"%{shared:crtbeginS.o%s;:crtbegin.o%s}"
 #endif
 
 #if ENABLE_VTABLE_VERIFY
@@ -228,7 +228,7 @@ along with GCC; see the file COPYING3.
 			       shared|" PIE_SPEC ":crtendS.o%s; \
 			       :crtend.o%s}"
 #else
-#define ENDFILE_CRTEND_SPEC "crtend.o%s"
+#define ENDFILE_CRTEND_SPEC "%{shared:crtendS.o%s;:crtend.o%s}"
 #endif
 
 #undef  ENDFILE_SPEC
diff -prwuN '--exclude=*.orig' gcc-7.3.0~/libgcc/config.host gcc-7.3.0/libgcc/config.host
--- gcc-7.3.0~/libgcc/config.host	2018-01-08 13:39:11.754860000 +0000
+++ gcc-7.3.0/libgcc/config.host	2018-05-08 14:44:26.269557826 +0000
@@ -270,7 +270,7 @@ case ${host} in
 *-*-solaris2*)
   # Unless linker support and dl_iterate_phdr are present,
   # unwind-dw2-fde-dip.c automatically falls back to unwind-dw2-fde.c.
-  tmake_file="$tmake_file sol2/t-sol2 t-eh-dw2-dip t-crtstuff-pic t-libgcc-pic t-slibgcc t-slibgcc-elf-ver"
+  tmake_file="$tmake_file sol2/t-sol2 t-eh-dw2-dip t-crtstuff-pic t-libgcc-pic t-slibgcc t-slibgcc-elf-ver t-crtstuff-pic"
   if test $with_gnu_ld = yes; then
     tmake_file="$tmake_file t-slibgcc-gld"
   else
@@ -290,6 +290,7 @@ case ${host} in
       i?86-*-solaris2.1[0-9]* | x86_64-*-solaris2.1[0-9]*)
         # Solaris 10+/x86 provides crt1.o, crti.o, crtn.o, and gcrt1.o as
         # part of the base system.
+        extra_parts="$extra_parts crtbeginS.o crtendS.o"
         ;;
       sparc*-*-solaris2.1[0-9]*)
         # Solaris 10+/SPARC lacks crt1.o and gcrt1.o.
