From f26041acf5d658a6bd58a4aca9ca15aae1efb1f4 Mon Sep 17 00:00:00 2001
From: Richard Lowe <richlowe@richlowe.net>
Date: Mon, 29 Oct 2018 18:21:34 +0000
Subject: [PATCH 15/17] libstdc++v3: illumos and Solaris haven't needed -lrt in
 a long time

Originally from Andy Fiddaman <andy@omniosce.org>
diff -wpruN '--exclude=*.orig' a~/libstdc++-v3/acinclude.m4 a/libstdc++-v3/acinclude.m4
--- a~/libstdc++-v3/acinclude.m4	1970-01-01 00:00:00
+++ a/libstdc++-v3/acinclude.m4	1970-01-01 00:00:00
@@ -1434,7 +1434,6 @@ AC_DEFUN([GLIBCXX_ENABLE_LIBSTDCXX_TIME]
         ac_has_nanosleep=yes
         ;;
       solaris*)
-        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
         ac_has_clock_monotonic=yes
         ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
diff -wpruN '--exclude=*.orig' a~/libstdc++-v3/configure a/libstdc++-v3/configure
--- a~/libstdc++-v3/configure	1970-01-01 00:00:00
+++ a/libstdc++-v3/configure	1970-01-01 00:00:00
@@ -20578,7 +20578,6 @@ $as_echo "$glibcxx_glibc217" >&6; }
         ac_has_nanosleep=yes
         ;;
       solaris*)
-        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
         ac_has_clock_monotonic=yes
         ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
