From 3bab576aebc4cce41169275744f57a112a0948b7 Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <omnios@citrus-it.co.uk>
Date: Fri, 19 Jul 2019 11:42:58 +0000
Subject: Use GNU backends for gcc-{ar,nm,ranlib}

---
 gcc/gcc-ar.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gcc/gcc-ar.c b/gcc/gcc-ar.c
index d5d80e042e5..5f1d075fdd9 100644
--- a/gcc/gcc-ar.c
+++ b/gcc/gcc-ar.c
@@ -187,12 +187,12 @@ main (int ac, char **av)
     }
 
   /* Find the wrapped binutils program.  */
-  exe_name = find_a_file (&target_path, PERSONALITY, X_OK);
+  exe_name = find_a_file (&target_path, "g" PERSONALITY, X_OK);
   if (!exe_name)
     {
-      const char *real_exe_name = PERSONALITY;
+      const char *real_exe_name = "g" PERSONALITY;
 #ifdef CROSS_DIRECTORY_STRUCTURE
-      real_exe_name = concat (target_machine, "-", PERSONALITY, NULL);
+      real_exe_name = concat (target_machine, "-", "g" PERSONALITY, NULL);
 #endif
       exe_name = find_a_file (&path, real_exe_name, X_OK);
       if (!exe_name)
-- 
2.21.0

