From edd50d59f54c13d99677a568eee19b530f0ef450 Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <omnios@citrus-it.co.uk>
Date: Wed, 1 Sep 2021 10:17:01 +0000
Subject: libstdc++ must use thread-local errno

---
 libstdc++-v3/configure    | 10 +++-------
 libstdc++-v3/configure.ac |  4 ++++
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/libstdc++-v3/configure b/libstdc++-v3/configure
index c57a30495cf..2e292f5ebe7 100755
--- a/libstdc++-v3/configure
+++ b/libstdc++-v3/configure
@@ -81332,13 +81332,9 @@ CPU_OPT_EXT_RANDOM=config/${cpu_opt_ext_random}
 CPU_OPT_BITS_RANDOM=config/${cpu_opt_bits_random}
 
 
-
-
-
-
-
-
-
+case "${target}" in *-*-solaris2*)
+        EXTRA_CXX_FLAGS="$EXTRA_CXX_FLAGS -D_TS_ERRNO"
+esac
 
 
 # Determine cross-compile flags and AM_CONDITIONALs.
diff --git a/libstdc++-v3/configure.ac b/libstdc++-v3/configure.ac
index b0243d1c188..99cc2dd2ad9 100644
--- a/libstdc++-v3/configure.ac
+++ b/libstdc++-v3/configure.ac
@@ -495,6 +495,10 @@ AC_SUBST(ERROR_CONSTANTS_SRCDIR)
 AC_SUBST(CPU_OPT_EXT_RANDOM)
 AC_SUBST(CPU_OPT_BITS_RANDOM)
 
+case "${target}" in *-*-solaris2*)
+	EXTRA_CXX_FLAGS="$EXTRA_CXX_FLAGS -D_TS_ERRNO"
+esac
+AC_SUBST(EXTRA_CXX_FLAGS)
 
 # Determine cross-compile flags and AM_CONDITIONALs.
 #AC_SUBST(GLIBCXX_IS_NATIVE)
