diff -prwuN '--exclude=*.orig' gcc-7.3.0~/libstdc++-v3/acinclude.m4 gcc-7.3.0/libstdc++-v3/acinclude.m4
--- gcc-7.3.0~/libstdc++-v3/acinclude.m4	2017-06-16 12:18:53.726871000 +0000
+++ gcc-7.3.0/libstdc++-v3/acinclude.m4	2018-05-08 14:43:36.619391807 +0000
@@ -1434,7 +1434,7 @@ AC_DEFUN([GLIBCXX_ENABLE_LIBSTDCXX_TIME]
         ac_has_nanosleep=yes
         ;;
       solaris*)
-        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
+#        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
         ac_has_clock_monotonic=yes
         ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
diff -prwuN '--exclude=*.orig' gcc-7.3.0~/libstdc++-v3/configure gcc-7.3.0/libstdc++-v3/configure
--- gcc-7.3.0~/libstdc++-v3/configure	2017-06-21 08:55:52.630537000 +0000
+++ gcc-7.3.0/libstdc++-v3/configure	2018-05-08 14:43:36.630504689 +0000
@@ -20576,7 +20576,7 @@ $as_echo "$glibcxx_glibc217" >&6; }
         ac_has_nanosleep=yes
         ;;
       solaris*)
-        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
+#        GLIBCXX_LIBS="$GLIBCXX_LIBS -lrt"
         ac_has_clock_monotonic=yes
         ac_has_clock_realtime=yes
         ac_has_nanosleep=yes
