diff -wpruN '--exclude=*.orig' a~/mesonbuild/compilers/compilers.py a/mesonbuild/compilers/compilers.py
--- a~/mesonbuild/compilers/compilers.py	1970-01-01 00:00:00
+++ a/mesonbuild/compilers/compilers.py	1970-01-01 00:00:00
@@ -1181,6 +1181,7 @@ class CompilerType(enum.Enum):
     GCC_OSX = 1
     GCC_MINGW = 2
     GCC_CYGWIN = 3
+    GCC_ILLUMOS = 4
 
     CLANG_STANDARD = 10
     CLANG_OSX = 11
@@ -1209,6 +1210,10 @@ class CompilerType(enum.Enum):
         return self.name in ('GCC_OSX', 'CLANG_OSX', 'ICC_OSX', 'PGI_OSX')
 
     @property
+    def is_illumos_compiler(self):
+        return self.name in ('GCC_ILLUMOS')
+
+    @property
     def is_windows_compiler(self):
         return self.name in ('GCC_MINGW', 'GCC_CYGWIN', 'CLANG_MINGW', 'ICC_WIN', 'ARM_WIN', 'CCRX_WIN', 'PGI_WIN')
 
diff -wpruN '--exclude=*.orig' a~/mesonbuild/environment.py a/mesonbuild/environment.py
--- a~/mesonbuild/environment.py	1970-01-01 00:00:00
+++ a/mesonbuild/environment.py	1970-01-01 00:00:00
@@ -591,7 +591,7 @@ class Environment:
         self.default_rust = ['rustc']
         self.default_swift = ['swiftc']
         self.default_vala = ['valac']
-        self.default_static_linker = ['ar', 'gar']
+        self.default_static_linker = ['gar']
         self.default_strip = ['strip']
         self.vs_static_linker = ['lib']
         self.clang_cl_static_linker = ['llvm-lib']
@@ -704,6 +704,8 @@ class Environment:
             return CompilerType.GCC_MINGW
         elif '__CYGWIN__' in defines:
             return CompilerType.GCC_CYGWIN
+        elif '__sun' in defines:
+            return CompilerType.GCC_ILLUMOS
         return CompilerType.GCC_STANDARD
 
     def _get_compilers(self, lang, for_machine):
