From 7de79ba32036c4b36945f1b1ba85f6094068412c Mon Jun 7 00:00:00 2001
From: Alexander Eremin <aeremin@tintri.com>
Date: Mon, 7 Jun 2021 12:09:48 +0000
Subject: Port SL_LOGONLY flag

diff -wpruN '--exclude=*.orig' a~/plugins/imsolaris/imsolaris.c a/plugins/imsolaris/imsolaris.c
--- a~/plugins/imsolaris/imsolaris.c	1970-01-01 00:00:00
+++ a/plugins/imsolaris/imsolaris.c	1970-01-01 00:00:00
@@ -92,6 +92,24 @@ MODULE_CNFNAME("imsolaris")
 /* defines */
 #define PATH_LOG	"/dev/log"
 
+/*
+ * Public flags for log messages
+ */
+#define	SL_FATAL	0x01    /* indicates fatal error */
+#define	SL_NOTIFY	0x02    /* logger must notify administrator */
+#define	SL_ERROR	0x04    /* include on the error log */
+#define	SL_TRACE	0x08    /* include on the trace log */
+#define	SL_CONSOLE	0x10    /* include on the console log */
+#define	SL_WARN		0x20    /* warning message */
+#define	SL_NOTE		0x40    /* notice message */
+
+/*
+ * Private flags for log messages -- used by internal implementation only
+ */
+#define	SL_CONSONLY	0x1000  /* send message only to /dev/console */
+#define	SL_LOGONLY	0x2000  /* send message only to /var/adm/messages */
+#define	SL_USER		0x4000  /* send message to user's terminal */
+#define	SL_PANICMSG	0x8000  /* message was created while panicking */
 
 /* Module static data */
 DEF_IMOD_STATIC_DATA
@@ -213,6 +231,8 @@ readLog(int fd, uchar *pRcv, int iMaxLin
 		MsgSetHOSTNAME(pMsg, glbl.GetLocalHostName(), ustrlen(glbl.GetLocalHostName()));
 		msgSetPRI(pMsg, hdr.pri);
 		pMsg->msgFlags = NEEDS_PARSING | NO_PRI_IN_RAW | IGNDATE;
+		if (hdr.flags & SL_LOGONLY)
+			pMsg->msgFlags |= LOGONLY;
 
 		/* Construct timestamp from msg ctl struct */
 		tim.tv_usec = 0;
diff -wpruN '--exclude=*.orig' a~/runtime/msg.h a/runtime/msg.h
--- a~/runtime/msg.h	1970-01-01 00:00:00
+++ a/runtime/msg.h	1970-01-01 00:00:00
@@ -158,6 +158,10 @@ struct msg {
 /* rawmsg does not include a PRI (Solaris!), but PRI is already set correctly in the msg object */
 #define PRESERVE_CASE	0x200
 /* preserve case in fromhost */
+#define LOGONLY			0x300
+/* send message only to /var/adm/messages */
+
+#define	SYSMSG		"/dev/sysmsg"
 
 /* (syslog) protocol types */
 #define MSG_LEGACY_PROTOCOL 0
diff -wpruN '--exclude=*.orig' a~/runtime/ruleset.c a/runtime/ruleset.c
--- a~/runtime/ruleset.c	1970-01-01 00:00:00
+++ a/runtime/ruleset.c	1970-01-01 00:00:00
@@ -586,6 +586,11 @@ scriptExec(struct cnfstmt *const root, s
 				  "force terminating\n");
 			ABORT_FINALIZE(RS_RET_FORCE_TERM);
 		}
+		if (stmt->printable != NULL &&
+		    (pMsg->msgFlags & LOGONLY) != 0 &&
+		    strcmp(stmt->printable, SYSMSG) == 0) {
+			goto finalize_it;
+		}
 		if(Debug) {
 			cnfstmtPrintOnly(stmt, 2, 0);
 		}
