diff -wpruN '--exclude=*.orig' a~/runtime/nsd_ossl.c a/runtime/nsd_ossl.c
--- a~/runtime/nsd_ossl.c	1970-01-01 00:00:00
+++ a/runtime/nsd_ossl.c	1970-01-01 00:00:00
@@ -28,6 +28,7 @@
 #include <stdlib.h>
 #include <assert.h>
 #include <string.h>
+#include <strings.h>
 #include <openssl/ssl.h>
 #include <openssl/x509v3.h>
 #include <openssl/err.h>
@@ -152,7 +153,7 @@ int opensslh_THREAD_setup(void)
 	for (i = 0; i < CRYPTO_num_locks( ); i++)
 		MUTEX_SETUP(mutex_buf[i]);
 
-	CRYPTO_set_id_callback(id_function);
+	//CRYPTO_set_id_callback(id_function);
 	CRYPTO_set_locking_callback(locking_function);
 	/* The following three CRYPTO_... functions are the OpenSSL functions
 	for registering the callbacks we implemented above */
@@ -173,7 +174,7 @@ int opensslh_THREAD_cleanup(void)
 	if (!mutex_buf)
 		return 0;
 
-	CRYPTO_set_id_callback(NULL);
+	//CRYPTO_set_id_callback(NULL);
 	CRYPTO_set_locking_callback(NULL);
 	CRYPTO_set_dynlock_create_callback(NULL);
 	CRYPTO_set_dynlock_lock_callback(NULL);
