diff -wpruN '--exclude=*.orig' a~/resize.man a/resize.man
--- a~/resize.man	1970-01-01 00:00:00
+++ a/resize.man	1970-01-01 00:00:00
@@ -45,7 +45,7 @@
 .el       .ds `` ``
 .ie \n(.g .ds '' \(rq
 .el       .ds '' ''
-.TH RESIZE 1 "__app_date__" "__app_version__" "X Window System"
+.TH RESIZE 1 "" "__app_version__" "X Window System"
 .SH NAME
 resize \- set environment and terminal settings to current xterm window size
 .SH SYNOPSIS
@@ -201,7 +201,7 @@ C shell syntax for the generated command
 TERM
 .IR \*N 's
 generated shell command
-sets this to "__default_termname__" if not already set.
+sets this to "xterm" if not already set.
 .TP 15
 TERMCAP
 .IR \*N 's
@@ -221,15 +221,9 @@ use those variables when set to override
 use_env(3x)
 .br
 csh(1), stty(1), tset(1)
-.br
-xterm(__mansuffix__)
 .SH AUTHORS
 Mark Vandevoorde (MIT-Athena), Edward Moy (Berkeley)
 .br
 Thomas Dickey (invisible-island.net).
 .br
 Copyright (c) 1984, 1985 by X Consortium
-.br
-See
-.IR X (__miscmansuffix__)
-for a complete copyright notice.
