diff -wpruN '--exclude=*.orig' a~/nspr/configure a/nspr/configure
--- a~/nspr/configure	1970-01-01 00:00:00
+++ a/nspr/configure	1970-01-01 00:00:00
@@ -7370,7 +7370,7 @@ $as_echo "$as_me: WARNING: Unknown versi
     # be linked against -lrt (or -lposix4) so add it to OS_LIBS
     _rev=`uname -r`
     _librt=`echo $_rev 5.6 | awk '{ if ($1 > $2) print "-lrt"; else print "-lposix4" }'`
-    OS_LIBS="$OS_LIBS $_librt"
+    # OS_LIBS="$OS_LIBS $_librt"
     ;;
 
 *-sco-sysv5*)
diff -wpruN '--exclude=*.orig' a~/nss/coreconf/SunOS5.mk a/nss/coreconf/SunOS5.mk
--- a~/nss/coreconf/SunOS5.mk	1970-01-01 00:00:00
+++ a/nss/coreconf/SunOS5.mk	1970-01-01 00:00:00
@@ -27,6 +27,10 @@ else
     else
       ARCHFLAG=-xarch=v8plus
     endif
+  else
+    ifdef NS_USE_GCC
+      ARCHFLAG=-m32
+    endif
   endif
 endif
 
@@ -38,7 +42,7 @@ ifdef NS_USE_GCC
 	OS_CFLAGS += -D__EXTENSIONS__
 	CCC        = g++
 	CCC       += -Wall -Wno-format
-	ASFLAGS	  += -x assembler-with-cpp
+	ASFLAGS	  += -x assembler-with-cpp $(ARCHFLAG)
 	OS_CFLAGS += $(NOMD_OS_CFLAGS) $(ARCHFLAG)
 	ifdef BUILD_OPT
 	    OPTIMIZER = -O2
@@ -103,6 +107,8 @@ PROCESS_MAP_FILE = grep -v ';-' $< | \
 ifdef NS_USE_GCC
 ifeq ($(USE_64), 1)
 	DSO_LDOPTS += -m64
+else
+	DSO_LDOPTS += -m32
 endif
 	DSO_LDOPTS += -shared -h $(notdir $@)
 else
@@ -139,4 +145,4 @@ else
 RPATH = -R '$$ORIGIN'
 endif
 
-OS_LIBS += -lthread -lnsl -lsocket -lposix4 -ldl -lc
+OS_LIBS += -lnsl -lsocket -ldl -lc
diff -wpruN '--exclude=*.orig' a~/nss/lib/freebl/Makefile a/nss/lib/freebl/Makefile
--- a~/nss/lib/freebl/Makefile	1970-01-01 00:00:00
+++ a/nss/lib/freebl/Makefile	1970-01-01 00:00:00
@@ -538,6 +538,8 @@ else
  	ifndef NS_USE_GCC
  	   MPCPU_SRCS =
  	   ASFILES += mpcpucache_x86.s
+ 	else
+ 	   ASFLAGS += -m32
  	endif
     endif
 endif # Solaris for non-sparc family CPUs
