diff -wpruN '--exclude=*.orig' a~/patch/pch.c a/patch/pch.c
--- a~/patch/pch.c	1970-01-01 00:00:00
+++ a/patch/pch.c	1970-01-01 00:00:00
@@ -42,7 +42,7 @@ static off_t	p_filesize;		/* size of the
 static LINENUM p_first;			/* 1st line number */
 static LINENUM p_newfirst;		/* 1st line number of replacement */
 static LINENUM p_ptrn_lines;		/* # lines in pattern */
-EXT    LINENUM p_repl_lines;		/* # lines in replacement text */
+extern LINENUM p_repl_lines;		/* # lines in replacement text */
 static LINENUM p_end = -1;		/* last line in hunk */
 static LINENUM p_max;			/* max allowed value of p_end */
 static LINENUM p_context = 3;		/* # of context lines */
