--- swig-4.0.2.clean/configure.ac	2020-06-08 19:19:47.000000000 +0000
+++ swig-4.0.2/configure.ac	2020-06-16 10:19:19.207479449 +0000
@@ -409,7 +409,7 @@
 esac
 
 # Check for specific libraries.   Used for SWIG examples
-AC_CHECK_LIB(dl, dlopen)	# Dynamic linking for SunOS/Solaris and SYSV
+#AC_CHECK_LIB(dl, dlopen)	# Dynamic linking for SunOS/Solaris and SYSV
 AC_CHECK_LIB(dld, shl_load)	# Dynamic linking for HP-UX
 
 dnl The following three libraries (nsl,inet,socket) are needed on Sequent,
@@ -419,9 +419,9 @@
 dnl Most SVR4 platforms will need -lsocket and -lnsl.  However on SGI IRIX 5,
 dnl these exist but are broken, so we use AC_SEARCH_LIBS which will only try
 dnl the library if the function isn't already available without it.
-AC_SEARCH_LIBS(t_open, nsl) # SVR4
+#AC_SEARCH_LIBS(t_open, nsl) # SVR4
 AC_SEARCH_LIBS(gethostbyname, inet) # Sequent
-AC_SEARCH_LIBS(socket, socket) # SVR4 sockets
+#AC_SEARCH_LIBS(socket, socket) # SVR4 sockets
 
 AC_CHECK_LIB(swill, swill_init, [SWIGLIBS="-lswill $LIBS" SWILL="-DSWIG_SWILL"])
 AC_SUBST(SWIGLIBS)
@@ -435,7 +435,7 @@
 then LIBM=$withval
 else AC_MSG_ERROR([proper usage is --with-libm=STRING])
 fi])
-AC_CHECK_LIB(ieee, main, [LIBM="-lieee $LIBM"])
+#AC_CHECK_LIB(ieee, main, [LIBM="-lieee $LIBM"])
 AC_CHECK_LIB(crypt,crypt, [LIBCRYPT="-lcrypt"])
 AC_SUBST(LIBCRYPT)
 
