diff -wpruN '--exclude=*.orig' a~/configure.ac a/configure.ac
--- a~/configure.ac	1970-01-01 00:00:00
+++ a/configure.ac	1970-01-01 00:00:00
@@ -465,7 +465,7 @@ case $host in
 esac
 
 # Check for specific libraries.   Used for SWIG examples
-AC_CHECK_LIB(dl, dlopen)	# Dynamic linking for SunOS/Solaris and SYSV
+#AC_CHECK_LIB(dl, dlopen)	# Dynamic linking for SunOS/Solaris and SYSV
 AC_CHECK_LIB(dld, shl_load)	# Dynamic linking for HP-UX
 
 dnl The following three libraries (nsl,inet,socket) are needed on Sequent,
@@ -475,9 +475,9 @@ dnl
 dnl Most SVR4 platforms will need -lsocket and -lnsl.  However on SGI IRIX 5,
 dnl these exist but are broken, so we use AC_SEARCH_LIBS which will only try
 dnl the library if the function isn't already available without it.
-AC_SEARCH_LIBS(t_open, nsl) # SVR4
+#AC_SEARCH_LIBS(t_open, nsl) # SVR4
 AC_SEARCH_LIBS(gethostbyname, inet) # Sequent
-AC_SEARCH_LIBS(socket, socket) # SVR4 sockets
+#AC_SEARCH_LIBS(socket, socket) # SVR4 sockets
 
 # check for --with-libm=...
 AC_SUBST(LIBM)
@@ -487,7 +487,7 @@ if test "$withval" != yes
 then LIBM=$withval
 else AC_MSG_ERROR([proper usage is --with-libm=STRING])
 fi])
-AC_CHECK_LIB(ieee, main, [LIBM="-lieee $LIBM"])
+#AC_CHECK_LIB(ieee, main, [LIBM="-lieee $LIBM"])
 AC_CHECK_LIB(crypt,crypt, [LIBCRYPT="-lcrypt"])
 AC_SUBST(LIBCRYPT)
 
