From 1c2ca31a404865bc4ebec31a7128f69f8270681c Mon Sep 17 00:00:00 2001
From: Dan McDonald <danmcd@joyent.com>
Date: Sat, 1 Jun 2019 13:54:13 -0400
Subject: [PATCH 02/10] Simplify hostname setting

---
 azurelinuxagent/common/osutil/illumos.py | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/azurelinuxagent/common/osutil/illumos.py b/azurelinuxagent/common/osutil/illumos.py
index 0cfd53e..d15936b 100644
--- a/azurelinuxagent/common/osutil/illumos.py
+++ b/azurelinuxagent/common/osutil/illumos.py
@@ -80,13 +80,10 @@ class illumosOSUtil(DefaultOSUtil):
         #
         fileutil.write_file('/etc/nodename', '{0}\n'.format(hostname))
 
-        ret = shellutil.run('svccfg -s svc:/system/identity:node setprop config/local_override = true')
+        # Make it happen NOW.
+        ret = shellutil.run('uname -S {0}'.format(hostname))
         if ret:
-            raise OSUtilError('Unable to set property of "svc:/system/identity:node" service.')
-
-        ret = shellutil.run('svcadm refresh svc:/system/identity:node')
-        if ret:
-            raise OSUtilError('Unable to refresh "svc:/system/identity:node" service.')
+            raise OSUtilError('Unable to set hostname to {0}.'.format(hostname))
 
         #
         # Unfortunately, there isn't a way to cause the service refresh
-- 
2.21.0

