From 361811beeee8d67b04b54f66e7c1a833ec0ef3ac Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <omnios@citrus-it.co.uk>
Date: Mon, 3 Jun 2019 14:38:46 +0000
Subject: [PATCH 03/12] Explicitly use python3

---
 bin/waagent    | 2 +-
 bin/waagent2.0 | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/bin/waagent b/bin/waagent
index 4039e03..7c6b7e6 100755
--- a/bin/waagent
+++ b/bin/waagent
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 #
 # Azure Linux Agent
 #
diff --git a/bin/waagent2.0 b/bin/waagent2.0
index 25aa0ce..a785daf 100644
--- a/bin/waagent2.0
+++ b/bin/waagent2.0
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 #
 # Azure Linux Agent
 #
-- 
2.23.0

