From 6a0068fecab63533cd3881ed7c0b4b7bb91db675 Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <omnios@citrus-it.co.uk>
Date: Thu, 6 Jun 2019 09:21:22 +0000
Subject: [PATCH 07/10] Reduce SCSI IO timeout to match OmniOS defaults

---
 config/illumos/waagent.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/illumos/waagent.conf b/config/illumos/waagent.conf
index 54f379f..6092004 100644
--- a/config/illumos/waagent.conf
+++ b/config/illumos/waagent.conf
@@ -16,7 +16,7 @@ Provisioning.AllowResetSysUser=n
 ResourceDisk.Format=n
 ResourceDisk.EnableSwap=n
 Logs.Verbose=y
-OS.RootDeviceScsiTimeout=300
+OS.RootDeviceScsiTimeout=10
 OS.EnableRDMA=n
 DetectScvmmEnv=n
 AutoUpdate.Enabled=n
-- 
2.21.0

