From 0bb70cae5a67d36984b339f00a600b0d874a45fd Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <omnios@citrus-it.co.uk>
Date: Thu, 6 Jun 2019 09:21:32 +0000
Subject: [PATCH 08/12] Disable verbose logs by default

---
 config/illumos/waagent.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/illumos/waagent.conf b/config/illumos/waagent.conf
index 6092004..1749267 100644
--- a/config/illumos/waagent.conf
+++ b/config/illumos/waagent.conf
@@ -15,7 +15,7 @@ Provisioning.ExecuteCustomData=n
 Provisioning.AllowResetSysUser=n
 ResourceDisk.Format=n
 ResourceDisk.EnableSwap=n
-Logs.Verbose=y
+Logs.Verbose=n
 OS.RootDeviceScsiTimeout=10
 OS.EnableRDMA=n
 DetectScvmmEnv=n
-- 
2.23.0

