From dc681d81f9ea760ee9f1a23e90f51bb91f7788c1 Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <omnios@citrus-it.co.uk>
Date: Thu, 6 Jun 2019 15:52:46 +0000
Subject: [PATCH 10/12] Pattern for sd_io_time needs relaxing

---
 azurelinuxagent/common/osutil/illumos.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/azurelinuxagent/common/osutil/illumos.py b/azurelinuxagent/common/osutil/illumos.py
index ecacbe6..3595d0b 100644
--- a/azurelinuxagent/common/osutil/illumos.py
+++ b/azurelinuxagent/common/osutil/illumos.py
@@ -238,7 +238,7 @@ class illumosOSUtil(DefaultOSUtil):
         return ret[1] if ret[0] == 0 else None
 
     def set_scsi_disks_timeout(self, timeout):
-        pattern = r'^set sd:sd_io_time = (.*)$'
+        pattern = r'^set sd:sd_io_time *= *(.*)$'
 
         #
         # Since changes to this setting require a reboot to take effect,
-- 
2.23.0

