From 3e802233c80d0746a293793b576690422d8829a7 Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <omnios@citrus-it.co.uk>
Date: Wed, 11 Dec 2019 12:07:36 +0000
Subject: [PATCH 11/12] Fix check for existing IO timeout, remove log messages

---
 azurelinuxagent/common/osutil/illumos.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/azurelinuxagent/common/osutil/illumos.py b/azurelinuxagent/common/osutil/illumos.py
index 3595d0b..77dc23b 100644
--- a/azurelinuxagent/common/osutil/illumos.py
+++ b/azurelinuxagent/common/osutil/illumos.py
@@ -252,17 +252,14 @@ class illumosOSUtil(DefaultOSUtil):
 
         for sf in ['/etc/system', '/etc/system.d/.self-assembly']:
                 if not os.path.isfile(sf): continue
-                match = fileutil.findstr_in_file(sf, pattern)
+                match = fileutil.findre_in_file(sf, pattern)
                 if match:
-                    logger.info('Found existing SCSI disk timeout setting: "{0}".'.format(match.group(0)))
-
                     try:
                         current = int(match.group(1))
                     except ValueError:
                         raise OSUtilError('Unable to parse existing SCSI disk timeout: "{0}".'.format(match.group(1)))
 
                     if current == int(timeout):
-                        logger.info('Current SCSI disk timeout matches desired SCSI disk timeout, skipping.')
                         return
 
         logger.warn('Updating SCSI disk timeout to desired value of "{0}", reboot required to take effect.'.format(timeout))
-- 
2.23.0

