From 9bc5d64787982373f759e28f3d6f50df42fa9345 Mon Sep 17 00:00:00 2001
From: Andy Fiddaman <omnios@citrus-it.co.uk>
Date: Wed, 28 Apr 2021 11:40:25 +0000
Subject: Add __illumos and __illumos__ preprocessor definitions

---
 gcc/config/sol2.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gcc/config/sol2.h b/gcc/config/sol2.h
index fb1e141127e..be47a318e74 100644
--- a/gcc/config/sol2.h
+++ b/gcc/config/sol2.h
@@ -85,6 +85,8 @@ along with GCC; see the file COPYING3.  If not see
   do {							\
     builtin_define_std ("unix");			\
     builtin_define_std ("sun");				\
+    builtin_define ("__illumos");			\
+    builtin_define ("__illumos__");			\
     builtin_define ("__svr4__");			\
     builtin_define ("__SVR4");				\
     builtin_assert ("system=unix");			\
-- 
2.31.0

