diff -wpruN '--exclude=*.orig' a~/snmplib/scapi.c a/snmplib/scapi.c
--- a~/snmplib/scapi.c	1970-01-01 00:00:00
+++ a/snmplib/scapi.c	1970-01-01 00:00:00
@@ -544,13 +544,11 @@ sc_hash(const oid * hashtype, size_t has
 
 #ifndef NETSNMP_DISABLE_MD5
     if (ISTRANSFORM(hashtype, HMACMD5Auth)) {
-	rval = pkcs_digest(CKM_MD5, buf, buf_len, MAC, &tmp_len);
-        *MAC_len = tmp_len;
+	rval = pkcs_digest(CKM_MD5, buf, buf_len, MAC, MAC_len);
     } else
 #endif
         if (ISTRANSFORM(hashtype, HMACSHA1Auth)) {
-	rval = pkcs_digest(CKM_SHA_1, buf, buf_len, MAC, &tmp_len);
-        *MAC_len = tmp_len;
+	rval = pkcs_digest(CKM_SHA_1, buf, buf_len, MAC, MAC_len);
     } else {
         return (SNMPERR_GENERR);
     }
