diff -wpruN '--exclude=*.orig' a~/agent/mibgroup/ucd-snmp/proxy.c a/agent/mibgroup/ucd-snmp/proxy.c
--- a~/agent/mibgroup/ucd-snmp/proxy.c	1970-01-01 00:00:00
+++ a/agent/mibgroup/ucd-snmp/proxy.c	1970-01-01 00:00:00
@@ -487,6 +487,8 @@ proxy_handler(netsnmp_mib_handler *handl
 
     /* Free any special parameters generated on the session */
     proxy_free_filled_in_session_args(sp->sess, (void **)&configured);
+    if (pdu)
+	snmp_free_pdu(pdu);
 
     return SNMP_ERR_NOERROR;
 }
