diff -wpruN '--exclude=*.orig' a~/snmplib/keytools.c a/snmplib/keytools.c
--- a~/snmplib/keytools.c	1970-01-01 00:00:00
+++ a/snmplib/keytools.c	1970-01-01 00:00:00
@@ -152,9 +152,7 @@ generate_Ku(const oid * hashtype, u_int
 #ifdef HAVE_EVP_MD_CTX_CREATE
     ctx = EVP_MD_CTX_create();
 #else
-    ctx = malloc(sizeof(*ctx));
-    if (!EVP_MD_CTX_init(ctx))
-        return SNMPERR_GENERR;
+   ctx = EVP_MD_CTX_new();
 #endif
 #ifndef NETSNMP_DISABLE_MD5
     if (ISTRANSFORM(hashtype, HMACMD5Auth)) {
@@ -262,8 +260,7 @@ generate_Ku(const oid * hashtype, u_int
 #ifdef HAVE_EVP_MD_CTX_DESTROY
         EVP_MD_CTX_destroy(ctx);
 #else
-        EVP_MD_CTX_cleanup(ctx);
-        free(ctx);
+	EVP_MD_CTX_free(ctx);
 #endif
     }
 #endif
diff -wpruN '--exclude=*.orig' a~/snmplib/scapi.c a/snmplib/scapi.c
--- a~/snmplib/scapi.c	1970-01-01 00:00:00
+++ a/snmplib/scapi.c	1970-01-01 00:00:00
@@ -489,12 +489,7 @@ sc_hash(const oid * hashtype, size_t has
 #ifdef HAVE_EVP_MD_CTX_CREATE
     cptr = EVP_MD_CTX_create();
 #else
-    cptr = malloc(sizeof(*cptr));
-#if defined(OLD_DES)
-    memset(cptr, 0, sizeof(*cptr));
-#else
-    EVP_MD_CTX_init(cptr);
-#endif
+    cptr = EVP_MD_CTX_new();
 #endif
     if (!EVP_DigestInit(cptr, hashfn)) {
         /* requested hash function is not available */
@@ -510,10 +505,7 @@ sc_hash(const oid * hashtype, size_t has
 #ifdef HAVE_EVP_MD_CTX_DESTROY
     EVP_MD_CTX_destroy(cptr);
 #else
-#if !defined(OLD_DES)
-    EVP_MD_CTX_cleanup(cptr);
-#endif
-    free(cptr);
+    EVP_MD_CTX_free(cptr);
 #endif
     return (rval);
 
