diff -wpruN '--exclude=*.orig' a~/configure a/configure
--- a~/configure	1970-01-01 00:00:00
+++ a/configure	1970-01-01 00:00:00
@@ -18628,7 +18628,7 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 fi
 
 if test "x$GCC" = "xyes"; then
-  CFLAGS="$CFLAGS -fexceptions"
+  CFLAGS="$CFLAGS -funwind-tables"
 fi
 
 cat > local.exp <<EOF
@@ -19826,7 +19826,7 @@ else $as_nop
 
   	libffi_cv_ro_eh_frame=yes
   	echo 'extern void foo (void); void bar (void) { foo (); foo (); }' > conftest.c
-  	if $CC $CFLAGS -c -fpic -fexceptions $libffi_cv_no_lto -o conftest.o conftest.c > /dev/null 2>&1; then
+  	if $CC $CFLAGS -c -fpic -funwind-tables $libffi_cv_no_lto -o conftest.o conftest.c > /dev/null 2>&1; then
 	    if $READELF -WS conftest.o | grep -q -n 'eh_frame .* WA'; then
 	        libffi_cv_ro_eh_frame=no
 	    fi
