diff -wpruN '--exclude=*.orig' a~/src/sys.h a/src/sys.h
--- a~/src/sys.h	1970-01-01 00:00:00
+++ a/src/sys.h	1970-01-01 00:00:00
@@ -115,7 +115,7 @@ typedef unsigned int	u_int32_t;
 #define	REGEX		/* Use POSIX.2 regular expression functions */
 #undef	REGEXP		/* Use UNIX V8 regular expression functions */
 
-#if defined(__sun)
+#if defined(__sun_NOT_REQUIRED)
 extern int tgetent(char *, const char *);
 extern int tgetflag(char *);
 extern int tgetnum(char *);
diff -wpruN '--exclude=*.orig' a~/src/terminal.h a/src/terminal.h
--- a~/src/terminal.h	1970-01-01 00:00:00
+++ a/src/terminal.h	1970-01-01 00:00:00
@@ -80,7 +80,7 @@ typedef struct {
 #define	A_K_DE		6
 #define	A_K_NKEYS	7
 
-#ifdef __sun
+#ifdef __sun_NOT_REQUIRED
 extern int tgetent(char *, const char *);
 extern int tgetflag(char *);
 extern int tgetnum(char *);
