diff -wpruN '--exclude=*.orig' a~/services/vmtoolsd/pluginMgr.c a/services/vmtoolsd/pluginMgr.c
--- a~/services/vmtoolsd/pluginMgr.c	1970-01-01 00:00:00
+++ a/services/vmtoolsd/pluginMgr.c	1970-01-01 00:00:00
@@ -654,7 +654,7 @@ ToolsCore_LoadPlugins(ToolsServiceState
    guint i;
    GPtrArray *plugins = NULL;
 
-#if defined(sun) && defined(__x86_64__)
+#if 0 && defined(sun) && defined(__x86_64__)
    const char *subdir = "/amd64";
 #else
    const char *subdir = "";
