diff -wpruN '--exclude=*.orig' a~/configure.ac a/configure.ac
--- a~/configure.ac	1970-01-01 00:00:00
+++ a/configure.ac	1970-01-01 00:00:00
@@ -1183,7 +1183,7 @@ else
    AC_CHECK_LIB(
       [pthread],
       [pthread_mutex_init],
-      [THREAD_LIBS=-lpthread],
+      [THREAD_LIBS=],
       [AC_MSG_ERROR(
          [libpthread not found. Please install the libc/libpthread devel package(s).])])
 fi
@@ -1618,7 +1618,7 @@ if test "$os" = "solaris"; then
    LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lnsl"
    LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lresolv"
    LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lrpcsvc"
-   LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lrt"
+   #LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lrt"
 
    # Setup defines to identify the OS version.
    if test "$osVersion" -eq 509000; then
