--- open-vm-tools-11.1.0-16036546~/configure.ac	2020-05-14 11:18:04.304138834 +0000
+++ open-vm-tools-11.1.0-16036546/configure.ac	2020-05-14 11:27:58.840409776 +0000
@@ -926,7 +926,7 @@
    AC_CHECK_LIB(
       [pthread],
       [pthread_mutex_init],
-      [THREAD_LIBS=-lpthread],
+      [THREAD_LIBS=],
       [AC_MSG_ERROR(
          [libpthread not found. Please install the libc/libpthread devel package(s).])])
 fi
@@ -1360,7 +1360,7 @@
    LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lnsl"
    LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lresolv"
    LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lrpcsvc"
-   LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lrt"
+   #LIBVMTOOLS_LIBADD="$LIBVMTOOLS_LIBADD -lrt"
 
    # Setup defines to identify the OS version.
    if test "$osVersion" -eq 509000; then
