
/dev/null _is_ pollable on illumos

diff -wpruN '--exclude=*.orig' a~/gio/tests/pollable.c a/gio/tests/pollable.c
--- a~/gio/tests/pollable.c	1970-01-01 00:00:00
+++ a/gio/tests/pollable.c	1970-01-01 00:00:00
@@ -383,7 +383,9 @@ main (int   argc,
   g_test_add_func ("/pollable/unix/pipe", test_pollable_unix_pipe);
   g_test_add_func ("/pollable/unix/pty", test_pollable_unix_pty);
   g_test_add_func ("/pollable/unix/file", test_pollable_unix_file);
+#ifndef __sun
   g_test_add_func ("/pollable/unix/nulldev", test_pollable_unix_nulldev);
+#endif
   g_test_add_func ("/pollable/converter", test_pollable_converter);
 #endif
   g_test_add_func ("/pollable/socket", test_pollable_socket);
