diff -wpruN '--exclude=*.orig' a~/meson.build a/meson.build
--- a~/meson.build	1970-01-01 00:00:00
+++ a/meson.build	1970-01-01 00:00:00
@@ -1912,6 +1912,8 @@ elif host_system == 'cygwin'
   export_dynamic_ldflags = ['-Wl,--export-all-symbols']
 elif host_system == 'darwin'
   export_dynamic_ldflags = []
+elif host_system == 'sunos'
+  export_dynamic_ldflags = []
 else
   export_dynamic_ldflags = ['-Wl,--export-dynamic']
 endif
diff -wpruN '--exclude=*.orig' a~/tests/meson.build a/tests/meson.build
--- a~/tests/meson.build	1970-01-01 00:00:00
+++ a/tests/meson.build	1970-01-01 00:00:00
@@ -45,8 +45,7 @@ tests = {
   'unicode-caseconv' : {},
   'unicode-encoding' : {},
   'module-test' : {
-    'dependencies' : [libgmodule_dep],
-    'export_dynamic' : true,
+    'dependencies' : [libgmodule_dep]
   },
   'cxx-test' : {
     'source' : 'cxx-test.cpp',
