diff -wpruN '--exclude=*.orig' a~/tests/meson.build a/tests/meson.build
--- a~/tests/meson.build	1970-01-01 00:00:00
+++ a/tests/meson.build	1970-01-01 00:00:00
@@ -45,14 +45,12 @@ tests = {
   'unicode-caseconv' : {},
   'unicode-encoding' : {},
   'module-test-library' : {
     'dependencies' : [libgmodule_dep],
-    'export_dynamic' : true,
     'source': 'module-test.c',
     'c_args': ['-DMODULE_TYPE="library"'],
   },
   'module-test-plugin' : {
     'dependencies' : [libgmodule_dep],
-    'export_dynamic' : true,
     'source': 'module-test.c',
     'c_args': ['-DMODULE_TYPE="plugin"'],
   },
