diff -wpruN '--exclude=*.orig' a~/meson.build a/meson.build
--- a~/meson.build	1970-01-01 00:00:00
+++ a/meson.build	1970-01-01 00:00:00
@@ -321,7 +321,6 @@ headers = [
   'sys/auxv.h',
   'sys/event.h',
   'sys/filio.h',
-  'sys/inotify.h',
   'sys/mkdev.h',
   'sys/mntctl.h',
   'sys/mnttab.h',
@@ -546,7 +545,6 @@ functions = [
   'getvfsstat',
   'gmtime_r',
   'hasmntopt',
-  'inotify_init1',
   'issetugid',
   'kevent',
   'kqueue',
