diff -wpruN '--exclude=*.orig' a~/libgomp/configure a/libgomp/configure
--- a~/libgomp/configure	1970-01-01 00:00:00
+++ a/libgomp/configure	1970-01-01 00:00:00
@@ -11486,7 +11486,7 @@ if { (eval echo "$as_me:$LINENO: \"$ac_c
   # Clean up.
   rm -f a.out a.exe
 else
-  echo "libtool.m4: error: problem compiling FC test program"
+  echo "libtool.m4: problem compiling FC test program"
 fi
 
 $RM -f confest.$objext
