diff -wpruN '--exclude=*.orig' a~/longlong.h a/longlong.h
--- a~/longlong.h	1970-01-01 00:00:00
+++ a/longlong.h	1970-01-01 00:00:00
@@ -1068,10 +1068,7 @@ extern UWtype __MPN(udiv_qrnnd) (UWtype
 	   : "=a" (q), "=d" (r)						\
 	   : "0" ((UDItype)(n0)), "1" ((UDItype)(n1)), "rm" ((UDItype)(dx)))
 
-#if HAVE_HOST_CPU_haswell || HAVE_HOST_CPU_broadwell || HAVE_HOST_CPU_skylake \
-  || HAVE_HOST_CPU_k10 || HAVE_HOST_CPU_bd1 || HAVE_HOST_CPU_bd2	\
-  || HAVE_HOST_CPU_bd3 || HAVE_HOST_CPU_bd4 || HAVE_HOST_CPU_zen	\
-  || HAVE_HOST_CPU_bobcat || HAVE_HOST_CPU_jaguar
+#ifdef ENABLE_BMI_INSTR
 #define count_leading_zeros(count, x)					\
   do {									\
     /* This is lzcnt, spelled for older assemblers.  Destination and */	\
@@ -1089,8 +1086,7 @@ extern UWtype __MPN(udiv_qrnnd) (UWtype
   } while (0)
 #endif
 
-#if HAVE_HOST_CPU_bd2 || HAVE_HOST_CPU_bd3 || HAVE_HOST_CPU_bd4 \
-  || HAVE_HOST_CPU_zen || HAVE_HOST_CPU_jaguar
+#ifdef ENABLE_BMI_INSTR
 #define count_trailing_zeros(count, x)					\
   do {									\
     /* This is tzcnt, spelled for older assemblers.  Destination and */	\
