--- gmp-6.2.0~/longlong.h	2020-01-17 13:44:49.000000000 +0000
+++ gmp-6.2.0/longlong.h	2020-03-04 19:17:57.599574039 +0000
@@ -1070,7 +1070,7 @@
 	   : "=a" (q), "=d" (r)						\
 	   : "0" ((UDItype)(n0)), "1" ((UDItype)(n1)), "rm" ((UDItype)(dx)))
 
-#if HAVE_HOST_CPU_haswell || HAVE_HOST_CPU_broadwell || HAVE_HOST_CPU_skylake \
+#if 0 && HAVE_HOST_CPU_haswell || HAVE_HOST_CPU_broadwell || HAVE_HOST_CPU_skylake \
   || HAVE_HOST_CPU_k10 || HAVE_HOST_CPU_bd1 || HAVE_HOST_CPU_bd2	\
   || HAVE_HOST_CPU_bd3 || HAVE_HOST_CPU_bd4 || HAVE_HOST_CPU_zen	\
   || HAVE_HOST_CPU_bobcat || HAVE_HOST_CPU_jaguar
