diff -ru ipmitool-1.8.18~/include/ipmitool/ipmi_hpmfwupg.h ipmitool-1.8.18/include/ipmitool/ipmi_hpmfwupg.h
--- ipmitool-1.8.18~/include/ipmitool/ipmi_hpmfwupg.h	2016-06-29 18:01:49.000000000 +0000
+++ ipmitool-1.8.18/include/ipmitool/ipmi_hpmfwupg.h	2020-05-27 12:02:14.062525573 +0000
@@ -800,8 +800,6 @@
 	char descString[HPMFWUPG_DESC_STRING_LENGTH + 1];
 }VERSIONINFO, *PVERSIONINFO;
 
-VERSIONINFO gVersionInfo[HPMFWUPG_COMPONENT_ID_MAX];
-
 #define TARGET_VER (0x01)
 #define ROLLBACK_VER (0x02)
 #define IMAGE_VER (0x04)
diff -ru ipmitool-1.8.18~/lib/ipmi_hpmfwupg.c ipmitool-1.8.18/lib/ipmi_hpmfwupg.c
--- ipmitool-1.8.18~/lib/ipmi_hpmfwupg.c	2020-05-27 11:59:25.825891858 +0000
+++ ipmitool-1.8.18/lib/ipmi_hpmfwupg.c	2020-05-27 12:02:10.676291895 +0000
@@ -53,6 +53,8 @@
 # include <config.h>
 #endif
 
+static VERSIONINFO gVersionInfo[HPMFWUPG_COMPONENT_ID_MAX];
+
 /* From src/plugins/ipmi_intf.c: */
 uint16_t
 ipmi_intf_get_max_request_data_size(struct ipmi_intf * intf);
