diff -wpruN '--exclude=*.orig' a~/src/plugins/ipmi_intf.c a/src/plugins/ipmi_intf.c
--- a~/src/plugins/ipmi_intf.c	1970-01-01 00:00:00
+++ a/src/plugins/ipmi_intf.c	1970-01-01 00:00:00
@@ -429,8 +429,11 @@ ipmi_intf_socket_connect(struct ipmi_int
 								tmp6->sin6_scope_id);
 						}
 
+#ifdef _sun
 						if (tmp6->sin6_scope_id != 0) {
+#endif /* _sun */
 							addr6->sin6_scope_id = tmp6->sin6_scope_id;
+#ifdef _sun
 						} else {
 							/*
 							 * No scope information in interface address information
@@ -444,6 +447,7 @@ ipmi_intf_socket_connect(struct ipmi_int
 								addr6->sin6_scope_id = ntohs(tmp6->sin6_addr.s6_addr[1]);
 							}
 						}
+#endif /* _sun */
 
 						/* OK, now try to connect with the scope id from this interface address */
 						if (addr6->sin6_scope_id != 0 || !IN6_IS_ADDR_LINKLOCAL(&tmp6->sin6_addr)) {
