--- src/runtime/syntax/dtrace.vim	2016-08-30 19:32:42.000000000 +0000
+++ src/runtime/syntax/dtrace.vim.new	2019-08-21 23:25:51.802064151 +0000
@@ -77,7 +77,7 @@
 
 " Turn reserved but unspecified keywords into errors
 syn keyword dtraceReservedKeyword auto break case continue counter default do
-syn keyword dtraceReservedKeyword else for goto if import probe provider
+syn keyword dtraceReservedKeyword for goto import probe provider
 syn keyword dtraceReservedKeyword register restrict return static switch while
 
 " Add dtrace-specific stuff
