--- a~/make/autoconf/flags-other.m4	Thu Jan 21 19:14:33 2021
+++ a/make/autoconf/flags-other.m4	Thu Jan 21 19:14:59 2021
@@ -116,8 +116,7 @@
 # $2 - Optional prefix for each variable defined.
 AC_DEFUN([FLAGS_SETUP_ASFLAGS_CPU_DEP],
 [
-  # Misuse EXTRA_CFLAGS to mimic old behavior
-  $2JVM_ASFLAGS="$JVM_BASIC_ASFLAGS ${$2EXTRA_CFLAGS}"
+  $2JVM_ASFLAGS="$JVM_BASIC_ASFLAGS ${$2EXTRA_ASFLAGS}"
 
   AC_SUBST($2JVM_ASFLAGS)
 ])
