diff -wpruN '--exclude=*.orig' a~/Makefile a/Makefile
--- a~/Makefile	1970-01-01 00:00:00
+++ a/Makefile	1970-01-01 00:00:00
@@ -29,7 +29,7 @@ OBJS= blocksort.o  \
 all: libbz2.a bzip2 bzip2recover test
 
 bzip2: libbz2.a bzip2.o
-	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2 bzip2.o -L. -lbz2
+	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2 bzip2.o -L. -lbz2 -lssp_ns
 
 bzip2recover: bzip2recover.o
 	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2recover bzip2recover.o
diff -wpruN '--exclude=*.orig' a~/Makefile-libbz2_so a/Makefile-libbz2_so
--- a~/Makefile-libbz2_so	1970-01-01 00:00:00
+++ a/Makefile-libbz2_so	1970-01-01 00:00:00
@@ -32,7 +32,7 @@ OBJS= blocksort.o  \
       bzlib.o
 
 all: $(OBJS)
-	$(CC) $(LDFLAGS) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.8 -nostdlib -lc $(OBJS)
+	$(CC) $(LDFLAGS) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.8 -nostdlib -lc $(OBJS) -lssp_ns
 	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.8
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.8 libbz2.so.1.0
