diff -wpruN '--exclude=*.orig' a~/client/dhclient.c a/client/dhclient.c
--- a~/client/dhclient.c	1970-01-01 00:00:00
+++ a/client/dhclient.c	1970-01-01 00:00:00
@@ -83,8 +83,8 @@ static const char message [] = "Internet
 static const char url [] = "For info, please visit https://www.isc.org/software/dhcp/";
 #endif /* UNIT_TEST */
 
-u_int16_t local_port = 0;
-u_int16_t remote_port = 0;
+extern u_int16_t local_port;
+extern u_int16_t remote_port;
 #if defined(DHCPv6) && defined(DHCP4o6)
 int dhcp4o6_state = -1; /* -1 = stopped, 0 = polling, 1 = started */
 #endif
diff -wpruN '--exclude=*.orig' a~/common/discover.c a/common/discover.c
--- a~/common/discover.c	1970-01-01 00:00:00
+++ a/common/discover.c	1970-01-01 00:00:00
@@ -45,8 +45,8 @@ struct interface_info *fallback_interfac
 
 int interfaces_invalidated;
 int quiet_interface_discovery;
-u_int16_t local_port;
-u_int16_t remote_port;
+u_int16_t local_port = 0;
+u_int16_t remote_port = 0;
 u_int16_t relay_port = 0;
 int dhcpv4_over_dhcpv6 = 0;
 int (*dhcp_interface_setup_hook) (struct interface_info *, struct iaddr *);
diff -wpruN '--exclude=*.orig' a~/relay/dhcrelay.c a/relay/dhcrelay.c
--- a~/relay/dhcrelay.c	1970-01-01 00:00:00
+++ a/relay/dhcrelay.c	1970-01-01 00:00:00
@@ -95,8 +95,8 @@ enum { forward_and_append,	/* Forward an
        forward_untouched,	/* Forward without changes. */
        discard } agent_relay_mode = forward_and_replace;
 
-u_int16_t local_port;
-u_int16_t remote_port;
+extern u_int16_t local_port;
+extern u_int16_t remote_port;
 
 /* Relay agent server list. */
 struct server_list {
diff -wpruN '--exclude=*.orig' a~/server/omapi.c a/server/omapi.c
--- a~/server/omapi.c	1970-01-01 00:00:00
+++ a/server/omapi.c	1970-01-01 00:00:00
@@ -47,7 +47,7 @@ omapi_object_type_t *dhcp_type_lease;
 omapi_object_type_t *dhcp_type_pool;
 omapi_object_type_t *dhcp_type_class;
 omapi_object_type_t *dhcp_type_subclass;
-omapi_object_type_t *dhcp_type_host;
+extern omapi_object_type_t *dhcp_type_host;
 #if defined (FAILOVER_PROTOCOL)
 omapi_object_type_t *dhcp_type_failover_state;
 omapi_object_type_t *dhcp_type_failover_link;
