diff -wpruN --no-dereference '--exclude=*.orig' a~/unix/configure a/unix/configure
--- a~/unix/configure	1970-01-01 00:00:00
+++ a/unix/configure	1970-01-01 00:00:00
@@ -588,9 +588,9 @@ rm -f a.out conftest.c conftest.o confte
 # bzip2
 
 echo "Check bzip2 support"
-D_USE_BZ2=""
+D_USE_BZ2="-DUSE_BZIP2"
 LIBBZ2=""
-L_BZ2=""
+L_BZ2="-lbz2"
 CC_BZ="${CC}"
 
 if test -n "${IZ_BZIP2}" -a "${IZ_BZIP2}" != "bzip2" ; then
