diff -wpruN '--exclude=*.orig' a~/unix/Makefile a/unix/Makefile
--- a~/unix/Makefile	1970-01-01 00:00:00
+++ a/unix/Makefile	1970-01-01 00:00:00
@@ -52,7 +52,7 @@ CF_NOOPT = -I. -I$(IZ_BZIP2) -DUNIX $(LO
 CF = $(CFLAGS) $(CF_NOOPT)
 LFLAGS1 =
 LF = -o unzip$E $(LFLAGS1)
-LF2 = -s
+LF2 =
 
 # UnZipSFX flags
 SL = -o unzipsfx$E $(LFLAGS1)
@@ -70,7 +70,7 @@ RM = rm -f
 CHMOD = chmod
 BINPERMS = 755
 MANPERMS = 644
-STRIP = strip
+STRIP = echo
 E =
 O = .o
 M = unix
diff -wpruN '--exclude=*.orig' a~/unix/configure a/unix/configure
--- a~/unix/configure	1970-01-01 00:00:00
+++ a/unix/configure	1970-01-01 00:00:00
@@ -17,7 +17,7 @@ CFLAGSR=${CFLAGS}
 IZ_BZIP2=${3}
 CFLAGS="${CFLAGS} -I. -DUNIX"
 LFLAGS1=""
-LFLAGS2="-s"
+LFLAGS2=""
 LN="ln -s"
 
 CFLAGS_OPT=''
