diff -wpruN '--exclude=*.orig' a~/unix/Makefile a/unix/Makefile
--- a~/unix/Makefile	1970-01-01 00:00:00
+++ a/unix/Makefile	1970-01-01 00:00:00
@@ -121,9 +121,9 @@ INSTALL_PROGRAM = $(INSTALL)
 INSTALL_D = mkdir -p
 # on some systems, manext=l and MANDIR=/usr/man/man$(manext) may be appropriate
 manext = 1
-prefix = /usr/local
+prefix = $(PREFIX)
 BINDIR = $(prefix)/bin#			where to install executables
-MANDIR = $(prefix)/man/man$(manext)#	where to install man pages
+MANDIR = $(prefix)/share/man/man$(manext)#	where to install man pages
 INSTALLEDBIN = $(BINDIR)/funzip$E $(BINDIR)/unzip$E $(BINDIR)/unzipsfx$E \
 	$(BINDIR)/zipgrep$E $(BINDIR)/zipinfo$E
 INSTALLEDMAN = $(MANDIR)/funzip.$(manext) $(MANDIR)/unzip.$(manext) \
