--- Python-3.7.7/setup.py	2020-06-27 21:06:30.440791050 +0000
+++ Python-3.7.7~/setup.py	2020-06-27 21:04:07.475186348 +0000
@@ -864,6 +864,9 @@
         elif self.compiler.find_library_file(lib_dirs, 'curses'):
             curses_library = 'curses'
 
+        if host_platform == 'sunos5':
+            curses_library = 'ncurses'
+
         if host_platform == 'darwin':
             os_release = int(os.uname()[2].split('.')[0])
             dep_target = sysconfig.get_config_var('MACOSX_DEPLOYMENT_TARGET')
