diff -wpruN '--exclude=*.orig' a~/Lib/test/test_gdb.py a/Lib/test/test_gdb.py
--- a~/Lib/test/test_gdb.py	1970-01-01 00:00:00
+++ a/Lib/test/test_gdb.py	1970-01-01 00:00:00
@@ -51,6 +51,9 @@ if gdb_major_version < 7:
                             % (gdb_major_version, gdb_minor_version,
                                gdb_version))
 
+if sys.platform.startswith("sunos"):
+    raise unittest.SkipTest("test doesn't work very well on Solaris")
+
 if not sysconfig.is_python_build():
     raise unittest.SkipTest("test_gdb only works on source builds at the moment.")
 
