diff -wpruN '--exclude=*.orig' Python-3.5.5~/configure.ac Python-3.5.5/configure.ac
--- Python-3.5.5~/configure.ac	2018-06-28 15:18:36.330041699 +0000
+++ Python-3.5.5/configure.ac	2018-06-28 15:18:38.122645210 +0000
@@ -1102,7 +1102,7 @@ if test $enable_shared = "yes"; then
           ;;
     SunOS*)
 	  LDLIBRARY='libpython$(LDVERSION).so'
-	  BLDLIBRARY='-Wl,-R,$(LIBDIR) -L. -lpython$(LDVERSION)'
+	  BLDLIBRARY='-L. -lpython$(LDVERSION)'
 	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
 	  INSTSONAME="$LDLIBRARY".$SOVERSION
 	  if test "$with_pydebug" != yes
