diff -wpruN '--exclude=*.orig' Python-3.5.5~/Lib/ctypes/util.py Python-3.5.5/Lib/ctypes/util.py
--- Python-3.5.5~/Lib/ctypes/util.py	2018-07-04 10:19:14.218126442 +0000
+++ Python-3.5.5/Lib/ctypes/util.py	2018-07-04 10:19:18.511078094 +0000
@@ -144,7 +144,7 @@ elif os.name == "posix":
                 return None
 
             try:
-                proc = subprocess.Popen(("/usr/ccs/bin/dump", "-Lpv", f),
+                proc = subprocess.Popen(("/usr/bin/dump", "-Lpv", f),
                                         stdout=subprocess.PIPE,
                                         stderr=subprocess.DEVNULL)
             except OSError:  # E.g. command not found
