diff -wpruN '--exclude=*.orig' Python-3.5.5~/setup.py Python-3.5.5/setup.py
--- Python-3.5.5~/setup.py	2018-07-04 10:19:12.048150057 +0000
+++ Python-3.5.5/setup.py	2018-07-04 10:19:19.053231414 +0000
@@ -1331,6 +1331,9 @@ class PyBuildExt(build_ext):
         curses_defines = []
         curses_includes = []
         panel_library = 'panel'
+        if host_platform == 'sunos5':
+            curses_library = 'ncurses'
+
         if curses_library == 'ncursesw':
             curses_defines.append(('HAVE_NCURSESW', '1'))
             curses_includes.append('/usr/include/ncursesw')
