diff -wpruN '--exclude=*.orig' Python-3.5.5~/setup.py Python-3.5.5/setup.py
--- Python-3.5.5~/setup.py	2018-07-04 10:19:19.053231414 +0000
+++ Python-3.5.5/setup.py	2018-07-04 10:19:19.587037657 +0000
@@ -814,8 +814,16 @@ class PyBuildExt(build_ext):
         exts.append( Extension('_posixsubprocess', ['_posixsubprocess.c']) )
 
         # socket(2)
+        socket_libs = []
+        if self.compiler.find_library_file(lib_dirs,'socket'):
+           socket_libs.append('socket')
+        if self.compiler.find_library_file(lib_dirs,'nsl'):
+           socket_libs.append('nsl')
+        if self.compiler.find_library_file(lib_dirs,'resolv'):
+           socket_libs.append('resolv')
         exts.append( Extension('_socket', ['socketmodule.c'],
-                               depends = ['socketmodule.h']) )
+                               depends = ['socketmodule.h'],
+                               libraries = socket_libs ) )
         # Detect SSL support for the socket module (via _ssl)
         search_for_ssl_incs_in = [
                               '/usr/local/ssl/include',
