diff -wpruN '--exclude=*.orig' openjdk-1.7.0~/hotspot/make/solaris/makefiles/sparcWorks.make openjdk-1.7.0/hotspot/make/solaris/makefiles/sparcWorks.make
--- openjdk-1.7.0~/hotspot/make/solaris/makefiles/sparcWorks.make	2018-05-29 17:08:46.033809598 +0000
+++ openjdk-1.7.0/hotspot/make/solaris/makefiles/sparcWorks.make	2018-05-29 17:12:04.203524382 +0000
@@ -30,13 +30,13 @@ ifeq ($(SPEC),)
 
   # Note that this 'as' is an older version of the Sun Studio 'fbe', and will
   #   use the older style options. The 'fbe' options will match 'cc' and 'CC'.
-  AS	= /usr/ccs/bin/as
+  AS	= $(UTILS_CCS_BIN_PATH)as
 
-  NM    = /usr/ccs/bin/nm
+  NM    = $(UTILS_CCS_BIN_PATH)nm
   NAWK  = /bin/nawk
 
-  MCS	= /usr/ccs/bin/mcs
-  STRIP	= /usr/ccs/bin/strip
+  MCS	= $(UTILS_CCS_BIN_PATH)mcs
+  STRIP	= $(UTILS_CCS_BIN_PATH)strip
 endif
 
 REORDER_FLAG = -xF
