diff -wpruN '--exclude=*.orig' openjdk-1.7.0~/jdk/make/common/shared/Defs-solaris.gmk openjdk-1.7.0/jdk/make/common/shared/Defs-solaris.gmk
--- openjdk-1.7.0~/jdk/make/common/shared/Defs-solaris.gmk	2018-05-29 17:11:35.932181389 +0000
+++ openjdk-1.7.0/jdk/make/common/shared/Defs-solaris.gmk	2018-05-29 17:12:11.318493194 +0000
@@ -134,6 +134,13 @@ else
   GCC_COMPILER_PATH = $(JDK_DEVTOOLS_DIR)/$(ARCH_FAMILY)/gnucc/bin/
 endif
 
+# OPENWIN_HOME: path to where the X11 environment is installed.
+#  NOTE: Must end with / so that it could be empty, allowing PATH usage.
+ifneq ($(ALT_OPENWIN_HOME),)
+  OPENWIN_HOME :=$(call PrefixPath,$(ALT_OPENWIN_HOME))
+else
+  OPENWIN_HOME  =/usr/X11R6/
+endif
 
 # Always build headless on Solaris
 BUILD_HEADLESS = true
diff -wpruN '--exclude=*.orig' openjdk-1.7.0~/jdk/make/sun/awt/Makefile openjdk-1.7.0/jdk/make/sun/awt/Makefile
--- openjdk-1.7.0~/jdk/make/sun/awt/Makefile	2018-05-29 17:11:35.981075601 +0000
+++ openjdk-1.7.0/jdk/make/sun/awt/Makefile	2018-05-29 17:12:11.318706699 +0000
@@ -655,6 +655,7 @@ CPPFLAGS += -I$(SHARE_SRC)/native/$(PKGD
 	    -I$(SHARE_SRC)/native/$(PKGDIR)/../dc/path \
             -I$(PLATFORM_SRC)/native/$(PKGDIR)/../jdga \
             -I$(PLATFORM_SRC)/native/$(PKGDIR) \
+            -I$(ALT_OPENWIN_HOME)/include \
 	    $(EVENT_MODEL)
 
 ifeq ($(PLATFORM), linux)
diff -wpruN '--exclude=*.orig' openjdk-1.7.0~/jdk/make/sun/awt/mawt.gmk openjdk-1.7.0/jdk/make/sun/awt/mawt.gmk
--- openjdk-1.7.0~/jdk/make/sun/awt/mawt.gmk	2018-05-29 17:11:35.968908305 +0000
+++ openjdk-1.7.0/jdk/make/sun/awt/mawt.gmk	2018-05-29 17:12:11.318815050 +0000
@@ -246,7 +246,8 @@ ifeq ($(PLATFORM), macosx)
 endif
 
 ifeq ($(PLATFORM), solaris)
-  CPPFLAGS += -I$(OPENWIN_HOME)/include/X11/extensions
+  CPPFLAGS += -I$(OPENWIN_HOME)/include/X11/extensions \
+               -I$(ALT_OPENWIN_HOME)/include
 endif
 
 ifeq ($(PLATFORM), macosx)
diff -wpruN '--exclude=*.orig' openjdk-1.7.0~/jdk/make/sun/font/Makefile openjdk-1.7.0/jdk/make/sun/font/Makefile
--- openjdk-1.7.0~/jdk/make/sun/font/Makefile	2018-05-29 17:11:35.978252345 +0000
+++ openjdk-1.7.0/jdk/make/sun/font/Makefile	2018-05-29 17:12:11.318977893 +0000
@@ -205,6 +205,7 @@ OTHER_LDLIBS  += -lawt $(LIBM) $(LIBCXX)
 else
 ifeq ($(PLATFORM), solaris)
   # Note that on Solaris, fontmanager is built against the headless library.
+  CFLAGS       += -I$(ALT_OPENWIN_HOME)/include
   LDFLAGS      += -L$(LIBDIR)/$(LIBARCH)/headless
   OTHER_LDLIBS += -lawt -L$(LIBDIR)/$(LIBARCH)/xawt -lmawt -lc $(LIBM) $(LIBCXX)
 else
diff -wpruN '--exclude=*.orig' openjdk-1.7.0~/jdk/make/sun/jawt/Makefile openjdk-1.7.0/jdk/make/sun/jawt/Makefile
--- openjdk-1.7.0~/jdk/make/sun/jawt/Makefile	2018-05-29 17:11:35.981699913 +0000
+++ openjdk-1.7.0/jdk/make/sun/jawt/Makefile	2018-05-29 17:12:11.319092275 +0000
@@ -123,7 +123,8 @@ ifeq ($(PLATFORM), solaris)
   ifndef BUILD_HEADLESS_ONLY
     OTHER_LDLIBS = -L$(LIBDIR)/$(LIBARCH) -L$(OPENWIN_LIB) -L$(LIBDIR)/$(LIBARCH)/xawt -lmawt -L/usr/openwin/sfw/lib$(ISA_DIR) -lXrender
   else
-    OTHER_LDLIBS = -L$(LIBDIR)/$(LIBARCH) -L$(OPENWIN_LIB) -L$(LIBDIR)/$(LIBARCH)/headless -lmawt -L/usr/openwin/sfw/lib$(ISA_DIR) -lXrender
+    OTHER_LDLIBS = -L$(LIBDIR)/$(LIBARCH) -lawt -L$(LIBDIR)/$(LIBARCH)/headless -lmawt
+    CFLAGS += -DHEADLESS
   endif
 endif # PLATFORM
 
diff -wpruN '--exclude=*.orig' openjdk-1.7.0~/jdk/src/share/native/sun/java2d/cmm/lcms/cmsopt.c openjdk-1.7.0/jdk/src/share/native/sun/java2d/cmm/lcms/cmsopt.c
--- openjdk-1.7.0~/jdk/src/share/native/sun/java2d/cmm/lcms/cmsopt.c	2018-05-29 17:11:37.613109856 +0000
+++ openjdk-1.7.0/jdk/src/share/native/sun/java2d/cmm/lcms/cmsopt.c	2018-05-29 17:12:11.319536821 +0000
@@ -1799,8 +1799,8 @@ cmsBool OptimizeMatrixShaper(cmsPipeline
         _cmsStageToneCurvesData* mpeC1 = (_cmsStageToneCurvesData*) cmsStageData(Curve1);
         _cmsStageToneCurvesData* mpeC2 = (_cmsStageToneCurvesData*) cmsStageData(Curve2);
 
-        // In this particular optimization, caché does not help as it takes more time to deal with
-        // the caché that with the pixel handling
+        // In this particular optimization, cache does not help as it takes more time to deal with
+        // the cache that with the pixel handling
         *dwFlags |= cmsFLAGS_NOCACHE;
 
         // Setup the optimizarion routines
diff -wpruN '--exclude=*.orig' openjdk-1.7.0~/jdk/src/share/native/sun/java2d/cmm/lcms/cmstypes.c openjdk-1.7.0/jdk/src/share/native/sun/java2d/cmm/lcms/cmstypes.c
--- openjdk-1.7.0~/jdk/src/share/native/sun/java2d/cmm/lcms/cmstypes.c	2018-05-29 17:11:37.573445186 +0000
+++ openjdk-1.7.0/jdk/src/share/native/sun/java2d/cmm/lcms/cmstypes.c	2018-05-29 17:12:11.320555837 +0000
@@ -1003,7 +1003,7 @@ cmsBool  Type_Text_Description_Write(str
     len = cmsMLUgetASCII(mlu, cmsNoLanguage, cmsNoCountry, NULL, 0);
 
     // Specification ICC.1:2001-04 (v2.4.0): It has been found that textDescriptionType can contain misaligned data
-    //(see clause 4.1 for the definition of “aligned”). Because the Unicode language
+    //(see clause 4.1 for the definition of aligned). Because the Unicode language
     // code and Unicode count immediately follow the ASCII description, their
     // alignment is not correct if the ASCII count is not a multiple of four. The
     // ScriptCode code is misaligned when the ASCII count is odd. Profile reading and
@@ -3120,10 +3120,10 @@ void Type_ColorantTable_Free(struct _cms
 //The namedColor2Type is a count value and array of structures that provide color
 //coordinates for 7-bit ASCII color names. For each named color, a PCS and optional
 //device representation of the color are given. Both representations are 16-bit values.
-//The device representation corresponds to the header’s “color space of data” field.
-//This representation should be consistent with the “number of device components”
+//The device representation corresponds to the headers color space of data field.
+//This representation should be consistent with the number of device components
 //field in the namedColor2Type. If this field is 0, device coordinates are not provided.
-//The PCS representation corresponds to the header’s PCS field. The PCS representation
+//The PCS representation corresponds to the headers PCS field. The PCS representation
 //is always provided. Color names are fixed-length, 32-byte fields including null
 //termination. In order to maintain maximum portability, it is strongly recommended
 //that special characters of the 7-bit ASCII set not be used.
@@ -3868,7 +3868,7 @@ void Type_Screening_Free(struct _cms_typ
 // ********************************************************************************
 //
 //This type represents a set of viewing condition parameters including:
-//CIE ’absolute’ illuminant white point tristimulus values and CIE ’absolute’
+//CIE absolute illuminant white point tristimulus values and CIE absolute
 //surround tristimulus values.
 
 static
@@ -3955,7 +3955,7 @@ void GenericMPEfree(struct _cms_typehand
 }
 
 // Each curve is stored in one or more curve segments, with break-points specified between curve segments.
-// The first curve segment always starts at –Infinity, and the last curve segment always ends at +Infinity. The
+// The first curve segment always starts at Infinity, and the last curve segment always ends at +Infinity. The
 // first and last curve segments shall be specified in terms of a formula, whereas the other segments shall be
 // specified either in terms of a formula, or by a sampled curve.
 
@@ -4236,7 +4236,7 @@ cmsBool  Type_MPEcurve_Write(struct _cms
 // The matrix is organized as an array of PxQ+Q elements, where P is the number of input channels to the
 // matrix, and Q is the number of output channels. The matrix elements are each float32Numbers. The array
 // is organized as follows:
-// array = [e11, e12, …, e1P, e21, e22, …, e2P, …, eQ1, eQ2, …, eQP, e1, e2, …, eQ]
+// array = [e11, e12, , e1P, e21, e22, , e2P, , eQ1, eQ2, , eQP, e1, e2, , eQ]
 
 static
 void *Type_MPEmatrix_Read(struct _cms_typehandler_struct* self, cmsIOHANDLER* io, cmsUInt32Number* nItems, cmsUInt32Number SizeOfTag)
@@ -4759,10 +4759,10 @@ void *Type_vcgt_Read(struct _cms_typehan
             // Y = cX + f             | X < d
 
             // vcgt formula is:
-            // Y = (Max – Min) * (X ^ Gamma) + Min
+            // Y = (Max  Min) * (X ^ Gamma) + Min
 
             // So, the translation is
-            // a = (Max – Min) ^ ( 1 / Gamma)
+            // a = (Max  Min) ^ ( 1 / Gamma)
             // e = Min
             // b=c=d=f=0
 
diff -wpruN '--exclude=*.orig' openjdk-1.7.0~/jdk/src/share/native/sun/java2d/cmm/lcms/cmsxform.c openjdk-1.7.0/jdk/src/share/native/sun/java2d/cmm/lcms/cmsxform.c
--- openjdk-1.7.0~/jdk/src/share/native/sun/java2d/cmm/lcms/cmsxform.c	2018-05-29 17:11:37.667393692 +0000
+++ openjdk-1.7.0/jdk/src/share/native/sun/java2d/cmm/lcms/cmsxform.c	2018-05-29 17:12:11.321012248 +0000
@@ -370,7 +370,7 @@ void NullFloatXFORM(_cmsTRANSFORM* p,
 
 // 16 bit precision -----------------------------------------------------------------------------------------------------------
 
-// Null transformation, only applies formatters. No caché
+// Null transformation, only applies formatters. No cache
 static
 void NullXFORM(_cmsTRANSFORM* p,
                const void* in,
@@ -471,7 +471,7 @@ void TransformOnePixelWithGamutCheck(_cm
         p ->Lut ->Eval16Fn(wIn, wOut, p -> Lut->Data);
 }
 
-// Gamut check, No caché, 16 bits.
+// Gamut check, No cache, 16 bits.
 static
 void PrecalculatedXFORMGamutCheck(_cmsTRANSFORM* p,
                                   const void* in,
@@ -510,7 +510,7 @@ void PrecalculatedXFORMGamutCheck(_cmsTR
 }
 
 
-// No gamut check, Caché, 16 bits,
+// No gamut check, Cache, 16 bits,
 static
 void CachedXFORM(_cmsTRANSFORM* p,
                  const void* in,
@@ -907,16 +907,16 @@ _cmsTRANSFORM* AllocEmptyTransform(cmsCo
             if (*dwFlags & cmsFLAGS_NOCACHE) {
 
                 if (*dwFlags & cmsFLAGS_GAMUTCHECK)
-                    p ->xform = PrecalculatedXFORMGamutCheck;  // Gamut check, no caché
+                    p ->xform = PrecalculatedXFORMGamutCheck;  // Gamut check, no cache
                 else
-                    p ->xform = PrecalculatedXFORM;  // No caché, no gamut check
+                    p ->xform = PrecalculatedXFORM;  // No cache, no gamut check
             }
             else {
 
                 if (*dwFlags & cmsFLAGS_GAMUTCHECK)
-                    p ->xform = CachedXFORMGamutCheck;    // Gamut check, caché
+                    p ->xform = CachedXFORMGamutCheck;    // Gamut check, cache
                 else
-                    p ->xform = CachedXFORM;  // No gamut check, caché
+                    p ->xform = CachedXFORM;  // No gamut check, cache
 
             }
         }
diff -wpruN '--exclude=*.orig' openjdk-1.7.0~/jdk/src/solaris/native/sun/awt/jawt.c openjdk-1.7.0/jdk/src/solaris/native/sun/awt/jawt.c
--- openjdk-1.7.0~/jdk/src/solaris/native/sun/awt/jawt.c	2018-05-29 17:11:37.865001763 +0000
+++ openjdk-1.7.0/jdk/src/solaris/native/sun/awt/jawt.c	2018-05-29 17:12:11.320679981 +0000
@@ -33,7 +33,7 @@
  */
 JNIEXPORT jboolean JNICALL JAWT_GetAWT(JNIEnv* env, JAWT* awt)
 {
-#if defined(JAVASE_EMBEDDED) && defined(HEADLESS)
+#if defined(JAVASE_EMBEDDED) || defined(HEADLESS)
     /* there are no AWT libs available at all */
     return JNI_FALSE;
 #else
