diff -wpruN '--exclude=*.orig' a~/configure a/configure
--- a~/configure	1970-01-01 00:00:00
+++ a/configure	1970-01-01 00:00:00
@@ -17396,7 +17396,7 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 fi
 
 if test "x$GCC" = "xyes"; then
-  CFLAGS="$CFLAGS -fexceptions"
+  CFLAGS="$CFLAGS -funwind-tables"
 fi
 
 cat > local.exp <<EOF
@@ -18547,7 +18547,7 @@ else
 
   	libffi_cv_ro_eh_frame=no
   	echo 'extern void foo (void); void bar (void) { foo (); foo (); }' > conftest.c
-  	if $CC $CFLAGS -c -fpic -fexceptions -o conftest.o conftest.c > /dev/null 2>&1; then
+  	if $CC $CFLAGS -c -fpic -funwind-tables -o conftest.o conftest.c > /dev/null 2>&1; then
 	    objdump -h conftest.o > conftest.dump 2>&1
 	    libffi_eh_frame_line=`grep -n eh_frame conftest.dump | cut -d: -f 1`
 	    if test "x$libffi_eh_frame_line" != "x"; then
